



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/cmdline_assembly_v2/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.2.0">
    
    
      
        <title>Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.572ca0f0.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.8c900955.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#long-read-assembly-workshop" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Long read assembly workshop
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy-data/" title="Download and Delete Data in Galaxy" class="md-nav__link">
      Download and Delete Data in Galaxy
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant calling - bacteria" class="md-nav__link">
      Variant calling - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - eukaryotes" class="md-nav__link">
      RNA-seq - eukaryotes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - Bacterial RNA-seq" class="md-nav__link">
      Workflow - Bacterial RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-metagenomics/" title="Workflow - Metagenomics" class="md-nav__link">
      Workflow - Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../help/" title="Help!" class="md-nav__link">
      Help!
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-get-started" title="1. Get started" class="md-nav__link">
    1. Get started
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mac-users" title="Mac users" class="md-nav__link">
    Mac users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#windows-users" title="Windows users" class="md-nav__link">
    Windows users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-new-working-directory-on-your-remote-computer" title="Create a new working directory on your remote computer." class="md-nav__link">
    Create a new working directory on your remote computer.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-get-data" title="2. Get data" class="md-nav__link">
    2. Get data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-assemble" title="3. Assemble" class="md-nav__link">
    3. Assemble
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-check-assembly-output" title="4. Check assembly output" class="md-nav__link">
    4. Check assembly output
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-trim-and-circularise" title="5. Trim and circularise" class="md-nav__link">
    5. Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output" title="Check the output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-find-smaller-plasmids" title="6. Find smaller plasmids" class="md-nav__link">
    6. Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-correct-the-assembly" title="7. Correct the assembly" class="md-nav__link">
    7. Correct the assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-comparative-genomics" title="8. Comparative Genomics" class="md-nav__link">
    8. Comparative Genomics
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assemblies" title="Assemblies" class="md-nav__link">
    Assemblies
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" title="Annotations" class="md-nav__link">
    Annotations
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#9-summary" title="9. Summary" class="md-nav__link">
    9. Summary
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/cmdline_assembly_v2/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="long-read-assembly-workshop">Long read assembly workshop</h1>
<p>This is a tutorial for a workshop on long-read (PacBio) genome assembly. </p>
<p>It demonstrates how to use long PacBio sequencing reads to assemble a bacterial genome, and includes additional steps for circularising, trimming, finding plasmids, and correcting the assembly with short-read Illumina data. </p>
<h2 id="overview">Overview</h2>
<p>Simplified version of workflow:</p>
<p><img alt="workflow" src="images/flowchart.png" /></p>
<h2 id="1-get-started">1. Get started</h2>
<p>Your workshop trainers will provide you with the address of a virtual machine. </p>
<h3 id="mac-users">Mac users</h3>
<p>Open the Terminal. </p>
<ul>
<li>Type in </li>
</ul>
<div class="codehilite"><pre><span></span>ssh researcher@[your virtual machine address]
</pre></div>


<ul>
<li>Type in the password provided. </li>
</ul>
<h3 id="windows-users">Windows users</h3>
<p>If you are using Windows 10, you might be able to use the Ubuntu Subsystem. Otherwise, install and open Putty. </p>
<ul>
<li>Download putty <a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html">here</a>.</li>
<li>Open. A configuration window will appear. </li>
<li>Under &ldquo;Host Name (or IP address)&rdquo; enter in the address of your virtual machine. </li>
<li>Under &ldquo;Port&rdquo; type in 22</li>
<li>Under &ldquo;Connection Type&rdquo; select &ldquo;SSH&rdquo;</li>
<li>Click &ldquo;Open&rdquo;</li>
<li>Under &ldquo;Login as:&rdquo; enter &ldquo;researcher&rdquo;</li>
<li>Type in the password provided. </li>
</ul>
<h3 id="create-a-new-working-directory-on-your-remote-computer">Create a new working directory on your remote computer.</h3>
<p>Because we are starting a new analysis it is always good practice to start in a new empty directory. Therefore, we will create a new directory and change to it for the rest of the workshop.</p>
<p>In your terminal:</p>
<ul>
<li>Create a new directory called &ldquo;Workshop&rdquo;</li>
</ul>
<div class="codehilite"><pre><span></span>mkdir Workshop
</pre></div>


<ul>
<li>Change to that directory</li>
</ul>
<div class="codehilite"><pre><span></span>cd Workshop
</pre></div>


<p><strong>NOTE: Everytime you open a new terminal or Putty session, you will need to make sure you are in this directory again.</strong></p>
<p>The current directory can be obtained with the linux command:</p>
<div class="codehilite"><pre><span></span>pwd
</pre></div>


<h2 id="2-get-data">2. Get data</h2>
<p>The sample used in this tutorial is from a bacteria called <em>Staphylococcus aureus</em>. We have used a small section of its real genome so that the programs can run in the workshop time. </p>
<p>The files we need are:</p>
<ul>
<li><fn>pacbio.fq</fn>: the PacBio reads</li>
<li><fn>R1.fq</fn>: the Illumina forward reads</li>
<li><fn>R2.fq</fn>: the Illumina reverse reads</li>
</ul>
<p>In a new tab, go to <a href="https://doi.org/10.5281/zenodo.1009308">https://doi.org/10.5281/zenodo.1009308</a>. </p>
<ul>
<li>Next to the first file, right-click (or control-click) the &ldquo;Download&rdquo; button, and select &ldquo;Copy link address&rdquo;.</li>
<li>Back in your terminal, enter </li>
</ul>
<div class="codehilite"><pre><span></span>wget [paste file link here]
</pre></div>


<ul>
<li>The file should download. </li>
<li><em>Note:</em> paste the link to the file, not to the webpage.</li>
<li>Repeat this for the other two files. </li>
</ul>
<h2 id="3-assemble">3. Assemble<a name="assemble"></a></h2>
<p>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>, version 1.6.</p>
<p>Run Canu with these commands:</p>
<div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir genomeSize=0.03m -pacbio-raw pacbio.fq
</pre></div>


<ul>
<li>the first <code>canu</code> tells the program to run</li>
<li><code>-p canu</code> names prefix for output files (&ldquo;canu&rdquo;)</li>
<li><code>-d canu_outdir</code> names output directory (&ldquo;canu_outdir&rdquo;)</li>
<li><code>genomeSize</code> only has to be approximate. (In this case we are using a partial genome of expected size 30,000 base pairs). </li>
<li>Canu will correct, trim and assemble the reads.</li>
<li>Various output will be displayed on the screen.</li>
<li><em>Note</em>: Canu could say &ldquo;Finished&rdquo; but may still be running. In this case, type <code>squeue</code> to see if jobs are still running. </li>
</ul>
<p>If you run <code>squeue</code> you will see something like this:</p>
<div class="codehilite"><pre><span></span>             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
                 6      main canu_can research PD       0:00      1 (Dependency)
               5_1      main cormhap_ research  R       0:29      1 master
</pre></div>


<p>You will know if <strong>Canu</strong> has completely finished when <code>squeue</code> shows no jobs listed under the header row. </p>
<h2 id="4-check-assembly-output">4. Check assembly output</h2>
<p>Move into the canu output folder: </p>
<div class="codehilite"><pre><span></span>cd canu_outdir
</pre></div>


<p>View the list of files: </p>
<div class="codehilite"><pre><span></span>ls
</pre></div>


<ul>
<li>The <fn>canu.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>canu.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>canu.correctedReads.fasta.gz</fn> are the corrected Pacbio reads that were used in the assembly.</li>
<li>The <fn>canu.contigs.gfa</fn> is the graph of the assembly.</li>
<li>The <fn>canu.report</fn> file is a summary of all of the steps Canu performed with information about the reads used, how they were handled and a whole lot of summary information about the assembly.</li>
</ul>
<p>Display summary information about the contigs: (<code>infoseq</code> is a tool from <a href="http://emboss.sourceforge.net/index.html">EMBOSS</a>)</p>
<div class="codehilite"><pre><span></span>infoseq canu.contigs.fasta
</pre></div>


<ul>
<li>This will show the contigs found by Canu. e.g.,  tig00000001  39136</li>
<li>&ldquo;tig00000001&rdquo; is the name given to the contig</li>
<li>&ldquo;39136&rdquo; is the number of base pairs in that contig.</li>
</ul>
<p>This matches what we were expecting for this sample (approximately 30,000 base pairs). For other data, Canu may not be able to join all the reads into one contig, so there may be several contigs in the output.</p>
<p>We should also look at the <fn>canu.report</fn>. To do this:</p>
<div class="codehilite"><pre><span></span>less canu.report
</pre></div>


<ul>
<li>&ldquo;less&rdquo; is a command to display the file on the screen.</li>
<li>Use the up and down arrows to scroll up and down. </li>
<li>You will see lots of histograms of read lengths before and after processing, final contig construction, etc. </li>
<li>For a description of the outputs that Canu produces, see: <a href="http://canu.readthedocs.io/en/latest/tutorial.html#outputs">http://canu.readthedocs.io/en/latest/tutorial.html#outputs</a></li>
<li>Type <code>q</code> to exit viewing the report. </li>
</ul>
<h3 id="questions">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do long- and short-read assembly methods differ?</p>
<details><summary>Answer</summary><p>Short reads are usually assembled with De Bruijn graphs. For long reads, there is a move back towards simpler overlap-layout-consensus methods.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where can we find out the what the approximate genome size should be for the species being assembled?</p>
<details><summary>Answer</summary><p>Go to NCBI Genomes, enter species name, click on Genome Assembly and Annotation report, sort table by clicking on the column header Size (Mb), look at range of sizes in this column.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>In the assembly output, what are the unassembled reads? Why are they there?</p>
<details><summary>Answer</summary><p>Reads and low-coverage contigs that were not used in the assembly. </p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are the corrected reads? How did canu correct the reads?</p>
<details><summary>Answer</summary><p>Canu builds overlaps between reads. The consensus is used to correct the reads. </p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where could you view the output .gfa and what would it show?</p>
<details><summary>Answer</summary><p>A useful program is <a href="https://rrwick.github.io/Bandage/">Bandage</a>. If the assembly has multiple contigs, the assembly graph shows how these are connected. </p>
</details>
</div>
<h2 id="5-trim-and-circularise">5. Trim and circularise</h2>
<p>Bacteria have circular chromosomes. </p>
<ul>
<li>Because of sequencing errors, there may be some &ldquo;overhang&rdquo; in the assembled linear sequence. </li>
<li>Our assembly may have some overhang because it is 9000 bases longer than expected. </li>
</ul>
<p><img alt="circlator" src="images/circlator_diagram.png" />
<em>Adapted from Figure 1. Hunt et al. Genome Biology 2015</em></p>
<p>A tool called <a href="http://sanger-pathogens.github.io/circlator/">Circlator</a> identifies and trims overhangs (on chromosomes and plasmids). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>Move back into your main analysis folder: </p>
<div class="codehilite"><pre><span></span>cd ..
</pre></div>


<h3 id="run-circlator">Run Circlator</h3>
<div class="codehilite"><pre><span></span>circlator all --threads 4 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir
</pre></div>


<ul>
<li>
<p>(Click on the dark grey slider bar above and move it to the right, to see all the way to the end of the line.)</p>
</li>
<li>
<p><code>--threads</code> is the number of cores</p>
</li>
<li><code>--verbose</code> prints progress information to the screen</li>
<li><code>canu_outdir/canu.contigs.fasta</code> is the file path to the input Canu assembly</li>
<li><code>canu_outdir/canu.correctedReads.fasta.gz</code> is the file path to the corrected Pacbio reads - note, fastA not fastQ</li>
<li><code>circlator_outdir</code> is the name of the output directory.</li>
</ul>
<p>Some output will print to screen. When finished, it should say &ldquo;Circularized x of x contig(s)&rdquo;.</p>
<h3 id="check-the-output">Check the output</h3>
<p>Move into the Circlator output directory: </p>
<div class="codehilite"><pre><span></span>cd circlator_outdir
</pre></div>


<p>List the files: </p>
<div class="codehilite"><pre><span></span>ls
</pre></div>


<p>Circlator has named the output files with numbers as prefixes. </p>
<p>Were the contigs circularised?</p>
<div class="codehilite"><pre><span></span>less 04.merge.circularise.log
</pre></div>


<ul>
<li>&ldquo;less&rdquo; is a command to display the file on the screen.</li>
<li><fn>04.merge.circularise.log</fn> is the name of the file. </li>
<li>Yes, the contig was circularised (last column).</li>
<li>Type <code>q</code> to exit.</li>
</ul>
<p>What are the trimmed contig sizes? </p>
<div class="codehilite"><pre><span></span>infoseq 06.fixstart.fasta
</pre></div>


<ul>
<li>The contig &ldquo;tig00000001&rdquo; has a length of 30019.</li>
<li>This is about 9000 bases shorter than before circularisation. This was the &ldquo;overhang&rdquo; and has now been trimmed. </li>
</ul>
<p>Copy the circularised contigs file to the main analysis directory with a new name:</p>
<div class="codehilite"><pre><span></span>cp 06.fixstart.fasta ../contig1.fasta
</pre></div>


<p>Move back into the main folder: </p>
<div class="codehilite"><pre><span></span>cd ..
</pre></div>


<h3 id="questions_1">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Were all the contigs circularised? </p>
<details><summary>Answer</summary><p>In this example, yes, the contig was circularised. </p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Circlator can set the start of the sequence at a particular gene. Which gene does it use? Is this appropriate for all contigs?</p>
<details><summary>Answer</summary><p>Circlator uses dnaA (if present) for the chromosomal contig. For other contigs, it uses a centrally-located gene. However, ideally, plasmids would be oriented on a gene such as repA. It is possible to provide a file to Circlator to do this.</p>
</details>
</div>
<h2 id="6-find-smaller-plasmids">6. Find smaller plasmids</h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the Canu+Circlator output of a trimmed assembly contig.</li>
<li>Map all the Illumina reads against this Pacbio-assembled contig.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim.</li>
</ol>
<h3 id="align-illumina-reads-to-the-pacbio-contig">Align Illumina reads to the PacBio contig</h3>
<p>Index the contigs file:</p>
<div class="codehilite"><pre><span></span>bwa index contig1.fasta
</pre></div>


<p>Align Illumina reads using using bwa mem:</p>
<div class="codehilite"><pre><span></span>bwa mem -t 4 contig1.fasta R1.fq R2.fq | samtools sort &gt; aln.bam
</pre></div>


<ul>
<li><code>bwa mem</code> is the alignment tool</li>
<li><code>-t 4</code> is the number of cores</li>
<li><code>contig1.fasta</code> is the input assembly file</li>
<li><code>R1.fq R2.fq</code> are the Illumina reads</li>
<li><code>| samtools sort</code> pipes the output to samtools to sort</li>
<li><code>&gt; aln.bam</code> sends the alignment to the file <fn>aln.bam</fn></li>
</ul>
<h3 id="extract-unmapped-illumina-reads">Extract unmapped Illumina reads</h3>
<p>Index the alignment file:</p>
<div class="codehilite"><pre><span></span>samtools index aln.bam
</pre></div>


<p>Extract the fastq files from the bam alignment - those reads that were unmapped to the Pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</p>
<div class="codehilite"><pre><span></span>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</pre></div>


<ul>
<li><code>fastq</code> is a command that coverts a <fn>.bam</fn> file into fastq format</li>
<li><code>-f 4</code> : only output unmapped reads</li>
<li><code>-1</code> : put R1 reads into a file called <fn>unmapped.R1.fastq</fn></li>
<li><code>-2</code> : put R2 reads into a file called <fn>unmapped.R2.fastq</fn></li>
<li><code>-s</code> : put singleton reads into a file called <fn>unmapped.RS.fastq</fn></li>
<li><code>aln.bam</code> : input alignment file</li>
</ul>
<p>We now have three files of the unampped reads: <fn> unmapped.R1.fastq</fn>, <fn> unmapped.R2.fastq</fn>, <fn> unmapped.RS.fastq</fn>.</p>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads</h3>
<p>Assemble with Spades:</p>
<div class="codehilite"><pre><span></span>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly
</pre></div>


<ul>
<li>
<p>(Click on the dark grey slider bar above and move it to the right, to see all the way to the end of the line.)</p>
</li>
<li>
<p><code>-1</code> is input file forward</p>
</li>
<li><code>-2</code> is input file reverse</li>
<li><code>-s</code> is unpaired</li>
<li><code>--careful</code> minimizes mismatches and short indels</li>
<li><code>--cov-cutoff auto</code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code>-o</code> is the output directory</li>
</ul>
<p>Move into the output directory: </p>
<div class="codehilite"><pre><span></span>cd spades_assembly
</pre></div>


<p>Look at the contigs: </p>
<div class="codehilite"><pre><span></span>infoseq contigs.fasta
</pre></div>


<ul>
<li>1 contig has been assembled with a length of 2359 bases. </li>
</ul>
<p>Copy it to a new file: </p>
<div class="codehilite"><pre><span></span>cp contigs.fasta contig2.fasta
</pre></div>


<h3 id="trim-the-plasmid">Trim the plasmid</h3>
<p>To trim any overhang on this plasmid, we will blast the start of contig2 against itself.</p>
<p>Take the start of the contig: </p>
<div class="codehilite"><pre><span></span>head -n 10 contig2.fasta &gt; contig2.fa.head
</pre></div>


<ul>
<li><code>head -n 10</code> takes the first ten lines of <fn>contig2.fasta</fn></li>
<li>
<p><code>&gt;</code> sends that output to a new file called <fn>contig2.fa.head</fn></p>
</li>
<li>
<p>We want to see if the start of the contig matches the end (overhang).</p>
</li>
</ul>
<p>Format the assembly file for blast: </p>
<div class="codehilite"><pre><span></span>makeblastdb -in contig2.fasta -dbtype nucl
</pre></div>


<ul>
<li><code>makeblastdb</code> makes a database for the tool Blast</li>
<li>This will generate three new files in the directory with suffixes .nhr, .nin and .nsq</li>
<li><code>-in</code> sets the input file as <fn> contig2.fasta</fn></li>
<li><code>-dbtype nucl</code> sets the type to nucleotide (rather than protein)</li>
</ul>
<p>Blast the start of the assembly (.head file) against all of the assembly: </p>
<div class="codehilite"><pre><span></span>blastn -query contig2.fa.head -db contig2.fasta -evalue 1e-3 -dust no -out contig2.bls
</pre></div>


<ul>
<li><code>blastn</code> is the tool Blast, set as blast<strong>n</strong> to compare sequences of nucleotides to each other</li>
<li><code>-query</code> sets the input sequence as <fn>contig2.fa.head</fn></li>
<li><code>-db</code> sets the database as that of the original sequence <fn>contig2.fasta</fn>. We don&rsquo;t have to specify the other files that were created when we formatted this file, but they need to present in our current directory. </li>
<li><code>-evalue</code> is the number of hits expected by chance, here set as 1e-3</li>
<li><code>-dust no</code> turns off the masking of low-complexity regions</li>
<li><code>-out</code> sets the output file as <fn>contig2.bls</fn></li>
</ul>
<p>Look at the hits (the matches): </p>
<div class="codehilite"><pre><span></span>less contig2.bls
</pre></div>


<ul>
<li>The first hit is at the start, as expected. We can see that &ldquo;Query 1&rdquo; (the start of the contig) is aligned to &ldquo;Sbject 1&rdquo; (the whole contig), for the first 540 bases.</li>
<li>Scroll down with the down arrow. </li>
<li>The second hit shows &ldquo;Query 1&rdquo; (the start of the contig) also matches to &ldquo;Sbject 1&rdquo; (the whole contig) at position 2253, all the way to the end, position 2359. </li>
</ul>
<p><img alt="blast" src="images/blasthits.png" /></p>
<ul>
<li>This is the overhang.</li>
<li>Therefore, in the next step, we need to trim the contig to position 2252.</li>
<li>Type <code>q</code> to exit. </li>
</ul>
<p>First, change the name of the contig within the file:</p>
<div class="codehilite"><pre><span></span>nano contig2.fasta
</pre></div>


<ul>
<li><code>nano</code> opens up a text editor. </li>
<li>Use the arrow keys to navigate. (The mouse won&rsquo;t work.)</li>
<li>At the first line, delete the text, which will be something like &ldquo;&gt;NODE_1_length_2359_cov_3.320333&rdquo;</li>
<li>Type in &ldquo;&gt;contig2&rdquo; </li>
<li>Don&rsquo;t forget the <code>&gt;</code> symbol</li>
<li>Press Control-X</li>
<li>&ldquo;Save modified buffer ?&rdquo; - type <code>Y</code></li>
<li>Press the Enter key</li>
</ul>
<p>Index the file (this will allow samtools to edit the file as it will have an index): </p>
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta
</pre></div>


<ul>
<li><code>faidx</code> means index the fasta file</li>
</ul>
<p>Trim the contig:</p>
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta contig2:1-2252 &gt; plasmid.fasta
</pre></div>


<ul>
<li>this extracts contig2 from position 1-2252</li>
<li>
<p><code>&gt; plasmid.fasta</code> sends the extracted section to a new file</p>
</li>
<li>
<p>We now have a trimmed plasmid.</p>
</li>
</ul>
<p>Copy the plasmid file into the main folder: </p>
<div class="codehilite"><pre><span></span>cp plasmid.fasta ../
</pre></div>


<p>Move file back into main folder: </p>
<div class="codehilite"><pre><span></span>cd ..
</pre></div>


<h3 id="collect-contigs">Collect contigs</h3>
<p>Collect the chromosome and the plasmid in one fasta file (they will be 2 records in the file): </p>
<div class="codehilite"><pre><span></span>cat contig1.fasta plasmid.fasta &gt; genome.fasta
</pre></div>


<p>See the contigs and sizes:</p>
<div class="codehilite"><pre><span></span>infoseq genome.fasta
</pre></div>


<ul>
<li>chromosome: 30019</li>
<li>plasmid: 2252</li>
</ul>
<h3 id="questions_2">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why is this section so complicated?</p>
<details><summary>Answer</summary><p>Finding small plasmids is difficult for many reasons! This paper has a nice summary: On the (im)possibility to reconstruct plasmids from whole genome short-read sequencing data. doi: https://doi.org/10.1101/086744</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can PacBio sequencing miss small plasmids?</p>
<details><summary>Answer</summary><p>Library prep size selection.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>We extract unmapped Illumina reads and assemble these to find small plasmids. What could they be missing?</p>
<details><summary>Answer</summary><p>Repeats that have mapped to the PacBio assembly.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do you find a plasmid in a Bandage graph?</p>
<details><summary>Answer</summary><p>It is probably circular, matches the size of a known plasmid, and has a rep gene.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Are there easier ways to find plasmids?</p>
<details><summary>Answer</summary><p>Possibly. One option is the program called Unicycler which may automate many of these steps. https://github.com/rrwick/Unicycler</p>
</details>
</div>
<h2 id="7-correct-the-assembly">7. Correct the assembly</h2>
<p>Sequences from PacBio can have more errors than those from Illumina. Therefore, although it is useful to use the long PacBio reads to assemble the genome, we can also use the shorter and more accurate Illumina reads to correct errors in the PacBio assembly. </p>
<h3 id="make-an-alignment-file">Make an alignment file</h3>
<p>Index the fasta file:</p>
<div class="codehilite"><pre><span></span>bwa index genome.fasta
</pre></div>


<p>Align the Illumina reads:</p>
<div class="codehilite"><pre><span></span>bwa mem -t 4 genome.fasta R1.fq R2.fq | samtools sort &gt; pilon_aln.bam
</pre></div>


<ul>
<li>Aligns Illumina <fn>R1.fq</fn> and <fn>R2.fq</fn> to the PacBio assembly <fn>genome.fasta</fn>. </li>
<li>This produces a .bam file</li>
<li><code>|</code> pipes the output to samtools to sort (required for downstream processing)</li>
<li><code>&gt; pilon_aln.bam</code> redirects the sorted bam to this file</li>
</ul>
<p>Index the files:</p>
<div class="codehilite"><pre><span></span>samtools index pilon_aln.bam
</pre></div>


<div class="codehilite"><pre><span></span>samtools faidx genome.fasta
</pre></div>


<ul>
<li>Now we have an alignment file to use  with the tool <a href="https://github.com/broadinstitute/pilon/wiki">Pilon</a>: <fn>pilon_aln.bam</fn></li>
</ul>
<h3 id="run-pilon">Run Pilon</h3>
<p>Run:</p>
<div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags pilon_aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --verbose --threads 4
</pre></div>


<ul>
<li><code>--genome</code> is the name of the input assembly to be corrected</li>
<li><code>--frags</code> is the alignment of the reads against the assembly</li>
<li><code>--output</code> is the name of the output prefix</li>
<li><code>--fix</code> is an option for types of corrections</li>
<li><code>--mindepth</code> gives a minimum read depth to use</li>
<li><code>--changes</code> produces an output file of the changes made</li>
<li><code>--verbose</code> prints information to the screen during the run</li>
<li><code>--threads</code>: number of cores</li>
</ul>
<p>Look at the changes file: </p>
<div class="codehilite"><pre><span></span>less pilon1.changes
</pre></div>


<p><em>Example:</em></p>
<p><img alt="pilon" src="images/pilon.png" /></p>
<ul>
<li>We can see lots of cases where a deletion (represented by a dot) has been corrected to a base.  </li>
<li>Type <code>q</code> to exit. </li>
</ul>
<p>Look at the details of the fasta file: </p>
<div class="codehilite"><pre><span></span>infoseq pilon1.fasta
</pre></div>


<ul>
<li>chromosome - 30060 (net +41 bases)</li>
<li>plasmid - 2252 (no change)</li>
</ul>
<p>Change the file name: </p>
<div class="codehilite"><pre><span></span>cp pilon1.fasta assembly.fasta
</pre></div>


<p>We now have the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing a chromosome and a small plasmid.  </p>
<h3 id="questions_3">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why don&rsquo;t we correct earlier in the assembly process?</p>
<details><summary>Answer</summary><p>We need to circularise the contigs and trim overhangs first.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can we use some reads (Illumina) to correct other reads (PacBio) ?</p>
<details><summary>Answer</summary><p>Illumina reads have higher accuracy.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Could we just use PacBio reads to assemble the genome?</p>
<details><summary>Answer</summary><p>Yes, if accuracy adequate.</p>
</details>
</div>
<h2 id="8-comparative-genomics">8. Comparative Genomics</h2>
<p>In the workshop so far, we used a partial bacterial genome so that the exercises could run in the time available. As a demonstration, to better see the effect of long and short reads on the assembly, we will examine complete bacterial genome. </p>
<h3 id="assemblies">Assemblies</h3>
<p>This bacterial genome has been assembled from either long PacBio reads (using Canu) or shorter Illumina reads (using Spades). </p>
<p>Assembly graphs:</p>
<p>Look at the assembly graph (usually has a suffix .gfa), in the program <a href="https://rrwick.github.io/Bandage/">Bandage</a>. This shows how contigs are related, albeit with ambiguity in some places.</p>
<p>The assembly graph from Illumina reads (Spades assembly):</p>
<p><img alt="spades assembly graph" src="images/25747_spades_illumina_graph.png" /></p>
<p>The assembly graph from PacBio reads (Canu assembly) - this is missing the small plasmid:</p>
<p><img alt="canu assembly graph" src="images/25747_canu_pacbio_graph.png" /></p>
<p>Here we can see that the long read data results in a more contiguous assembly - one complete chromosome versus many smaller contigs with ambiguous placement. </p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Does it matter that an assembly is in many contigs?</p>
<details><summary>Answer</summary><p>Yes and No. Yes: broken genes can lead to missing/incorrect annotations; fragmented assemblies provide less information about the genomic structure (<em>e.g.</em> the number of plasmids) and the location of genes of interest (<em>e.g.</em> gene A is located on plasmid X). No: many or all genes may still be annotated correctly. Gene location is useful (e.g. chromosome, plasmid1) but not always essential (e.g. presence/absence of particular resistance genes may be enough information).</p>
</details>
</div>
<h3 id="annotations">Annotations</h3>
<p>Genomic features such as genes can be identified with annotation tools. We have used a tool called <a href="https://github.com/tseemann/prokka">Prokka</a> to annotate the two genomes described above. </p>
<p>Some of the output data is displayed here:</p>
<p><center></p>
<table>
<thead>
<tr>
<th>assembly:</th>
<th align="right">PacBio</th>
<th align="right">Illumina</th>
</tr>
</thead>
<tbody>
<tr>
<td>size</td>
<td align="right">2,825,804</td>
<td align="right">2,792,905</td>
</tr>
<tr>
<td>contigs</td>
<td align="right">2</td>
<td align="right">123</td>
</tr>
<tr>
<td>CDS</td>
<td align="right">2614</td>
<td align="right">2575</td>
</tr>
<tr>
<td>tRNA</td>
<td align="right">61</td>
<td align="right">65</td>
</tr>
<tr>
<td>rRNA</td>
<td align="right">19</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<p></center></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why are there more CDS identified in the PacBio assembly? </p>
<details><summary>Answer</summary><p>The PacBio assembly may have errors (usually a one base indel) which will cause a frame shift, which can result in three things: a longer CDS, a shorter CDS, or a shorter CDS plus an additional CDS. In addition, the Illumina assembly is about 33 kb smaller than the PacBio assembly. In bacteria, a rule of thumb is that 1 kb is roughly equal to one gene. Thus, we would probably expect about 33 fewer identified genes, which fits with these results.  </p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why are there more rRNA identified in the PacBio assembly? </p>
<details><summary>Answer</summary><p>There may be multiple copies of the rRNAs and these could have been collapsed as repeats in the Illumina assembly. </p>
</details>
</div>
<h2 id="9-summary">9. Summary</h2>
<p>In this workshop, we used bacterial sequencing data from long and short reads to produce a polished genome. </p>
<p>Procedure and tools: </p>
<ul>
<li>Canu to assemble long-read PacBio data</li>
<li>Circlator to trim and circularise contigs</li>
<li>BWA-MEM to map shorter Illumina reads to the PacBio assembly</li>
<li>Spades to assemble any unmapped, leftover Illumina reads (the plasmid)</li>
<li>Pilon to correct the PacBio assembly with the more accurate Illumina reads</li>
</ul>
<p>We also looked at comparative genomics:</p>
<ul>
<li>Bandage to examine assembly graphs</li>
<li>Prokka to annotate genomes with features such as genes</li>
</ul>
<p>Further research:</p>
<ul>
<li>Align genomes with Mauve: <a href="http://sepsis-omics.github.io/tutorials/modules/mauve/">tutorial link</a></li>
<li>Find core and pan genomes with Roary and Phandango: <a href="http://sepsis-omics.github.io/tutorials/modules/roary/">tutorial link</a></li>
</ul>
<p>Melbourne Bioinformatics tutorials:</p>
<ul>
<li>https://www.melbournebioinformatics.org.au/tutorials/</li>
</ul>
<p>Additional microbial genomics tutorials:</p>
<ul>
<li>http://sepsis-omics.github.io/tutorials/</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b41f3d20.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>