



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/scripts/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.6">
    
    
      
        <title>Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.451f80e5.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#bash-scripts" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Home
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy-data/" title="Download and Delete Data in Galaxy" class="md-nav__link">
      Download and Delete Data in Galaxy
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant calling - bacteria" class="md-nav__link">
      Variant calling - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - eukaryotes" class="md-nav__link">
      RNA-seq - eukaryotes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - Bacterial RNA-seq" class="md-nav__link">
      Workflow - Bacterial RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-metagenomics/" title="Workflow - Metagenomics" class="md-nav__link">
      Workflow - Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#prepare-the-bash-script" title="Prepare the bash script" class="md-nav__link">
    Prepare the bash script
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#edit-the-bash-script" title="Edit the bash script" class="md-nav__link">
    Edit the bash script
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run" title="Run" class="md-nav__link">
    Run
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/scripts/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <p><br></p>
<h1 id="bash-scripts">Bash scripts</h1>
<p>It is possible to put some commands into a script, to run automatically.</p>
<p>Here, we will put commands for several tools into a bash script and use it to assemble a bacterial genome.</p>
<p>Bash is a programming language, and we can run a bash script in the bash shell (e.g. the terminal).</p>
<h2 id="prepare-the-bash-script">Prepare the bash script</h2>
<ul>
<li>Open the terminal.</li>
<li>Open a text editor, e.g., nano.</li>
</ul>
<div class="codehilite"><pre><span></span>nano myscript.sh
</pre></div>


<ul>
<li>We now have a script called myscript.sh</li>
<li>In the script, write</li>
</ul>
<div class="codehilite"><pre><span></span><span class="ch">#!/bin/bash  </span>
<span class="nb">echo</span> This is my assembly script
</pre></div>


<ul>
<li>
<p>Ctrl-X to exit nano.</p>
</li>
<li>
<p>Make the script executable:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span>chmod <span class="m">755</span> myscript.sh
</pre></div>


<ul>
<li>Run the script to test.</li>
</ul>
<div class="codehilite"><pre><span></span>./myscript.sh
</pre></div>


<ul>
<li>The text should print to the screen:</li>
</ul>
<div class="codehilite"><pre><span></span>This is my assembly script
</pre></div>


<h2 id="edit-the-bash-script">Edit the bash script</h2>
<ul>
<li>Now we will add to the bash script.</li>
<li>This will be a simple script that is designed to assemble, trim, and correct.</li>
<li>It will likely only work in simple cases.</li>
<li>Open nano</li>
</ul>
<div class="codehilite"><pre><span></span>nano myscript.sh
</pre></div>


<ul>
<li>and type in the following
(#comments and echo statements are optional but probably useful):</li>
</ul>
<div class="codehilite"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#a script for assembling a bacterial genome</span>

<span class="c1">#set a variable for the number of CPUS</span>
<span class="c1">#this can then be easily changed if needed</span>
<span class="nv">CPUS</span><span class="o">=</span><span class="m">16</span>

<span class="c1">#this script requires 5 arguments</span>
<span class="c1">#we will test whether 5 were entered</span>
<span class="k">if</span> <span class="o">[</span> ! <span class="nv">$#</span> -eq <span class="m">5</span> <span class="o">]</span> <span class="c1">#if there are not 5 arguments</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;ERROR: needs 5 arguments&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;usage: </span><span class="nv">$0</span><span class="s2"> outdir genomeSize (e.g. 2.1m) pacbio.fq R1.fq R2.fq&quot;</span>
    <span class="c1">#the usage shows what arguments are required</span>
    <span class="nb">exit</span> <span class="m">1</span> <span class="c1">#exits</span>
<span class="k">fi</span>

<span class="c1">#report back the arguments entered:</span>
<span class="c1">#each argument is assigned a variable name from 1 to 5</span>
<span class="c1">#these variables are referred to with the $ sign</span>
<span class="nb">echo</span> <span class="s2">&quot;your directory with output is called </span><span class="nv">$1</span><span class="s2">&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your genomeSize is </span><span class="nv">$2</span><span class="s2">&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your pacbio reads are in file </span><span class="nv">$3</span><span class="s2">&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your R1 reads are in file </span><span class="nv">$4</span><span class="s2">&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your R2 reads are in file </span><span class="nv">$5</span><span class="s2">&quot;</span>
<span class="nb">echo</span> <span class="c1">#echo on its own puts in a blank line on the screen</span>

<span class="c1">#run canu to assemble the reads</span>
mkdir -p <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">&quot;</span>  <span class="c1">#the p creates intermediate directories where required</span>
<span class="nb">echo</span> <span class="s2">&quot;now running canu&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your canu output is in directory &quot;</span><span class="nv">$1</span>/canu<span class="s2">&quot;&quot;</span>
nice canu -p canu -d <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu&quot;</span> <span class="nv">genomeSize</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$2</span><span class="s2">&quot;</span> -pacbio-raw <span class="s2">&quot;</span><span class="nv">$3</span><span class="s2">&quot;</span>
<span class="c1">#commands are preceeded by &#39;nice&#39; in this script for particular servers</span>

<span class="c1">#test canu output: number of contigs</span>
<span class="c1">#make a variable called num_contigs</span>
<span class="nv">num_contigs</span><span class="o">=</span><span class="k">$(</span>grep -c <span class="s1">&#39;&gt;&#39;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.contigs.fasta&quot;</span><span class="k">)</span>
<span class="nb">echo</span> <span class="s2">&quot;number of contigs found by canu is: </span><span class="nv">$num_contigs</span><span class="s2">&quot;</span>
<span class="nb">echo</span>
<span class="k">if</span> <span class="o">[</span> <span class="nv">$num_contigs</span> -gt <span class="m">10</span> <span class="o">]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;there are more than 10 contigs&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;the analysis is stopping&quot;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="c1">#in more complex scripts, you can add the option to repeat the analysis with different parameters, etc.</span>
<span class="k">fi</span>

<span class="c1">#run circlator to trim the contigs</span>
<span class="nb">echo</span> <span class="s2">&quot;now running circlator to trim and orient contigs&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your circlator output is in directory &quot;</span><span class="nv">$1</span>/circlator<span class="s2">&quot; &quot;</span>
nice circlator all --threads <span class="nv">$CPUS</span> --verbose <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.contigs.fasta&quot;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/canu/canu.correctedReads.fasta.gz&quot;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/circlator&quot;</span>
<span class="c1">#in more complex scripts, you can add information about which contigs were merged; where contigs were oriented; how contigs were trimmed, etc.</span>
<span class="c1">#rename the contigs file</span>
cp <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/circlator/06.fixstart.fasta&quot;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;your contigs from circlator are now in the file called contigs.fa&quot;</span>
<span class="c1">#note: in this case, these contigs may not have been trimmed.</span>

<span class="c1">#find smaller plasmids</span>
<span class="nb">echo</span> <span class="s2">&quot;now looking in illumina reads for small plasmids&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;indexing the contigs.fa file&quot;</span>
nice bwa index <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;aligning illumina reads to the contigs.fa file&quot;</span>
nice bwa mem -t <span class="nv">$CPUS</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span> <span class="s2">&quot;</span><span class="nv">$4</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$5</span><span class="s2">&quot;</span> <span class="p">|</span> samtools sort &gt; <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;indexing the aln.bam file&quot;</span>
nice samtools index <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;extracting the unmapped illumina reads&quot;</span>
samtools fastq -f <span class="m">4</span> -1 <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R1.fastq&quot;</span> -2 <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R2.fastq&quot;</span> -s <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.RS.fastq&quot;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;assembling the unmapped reads with spades&quot;</span>
<span class="nb">echo</span>
nice spades.py -t <span class="nv">$CPUS</span> -1 <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R1.fastq&quot;</span> -2 <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.R2.fastq&quot;</span> -s <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/unmapped.RS.fastq&quot;</span> -o <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/spades_assembly&quot;</span>
<span class="nb">echo</span>
<span class="nb">echo</span> <span class="s2">&quot;your output is in the spades_assembly directory&quot;</span>
<span class="nb">echo</span>

<span class="c1">#check if any contigs are &gt; 1500bp</span>
seqtk seq -L <span class="m">1500</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/spades_assembly/contigs.fasta&quot;</span> &gt; <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/illumina_contigs.fasta&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;extracting contigs &gt; 1500bp into illumina_contigs.fasta&quot;</span>
<span class="nb">echo</span>
<span class="c1">#in more complex scripts, you can trim and orient these contigs, or do further investigation into whether they are true plasmids</span>

<span class="c1">#join the canu pacbio contigs and the small plasmids</span>
cat <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/contigs.fa&quot;</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/illumina_contigs.fasta&quot;</span> &gt; <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>

<span class="nb">echo</span>
<span class="nb">echo</span> <span class="s2">&quot;joining small contigs to the pacbio contigs as allcontigs.fasta&quot;</span>
<span class="nb">echo</span>

<span class="c1">#correct this assembly with Pilon</span>
<span class="nb">echo</span> <span class="s2">&quot;preparing to correct contigs with illumina reads, using Pilon&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;indexing allcontigs.fasta&quot;</span>
nice bwa index <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;aligning illumina reads to allcontigs.fasta&quot;</span>
nice bwa mem -t <span class="nv">$CPUS</span> <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span> <span class="s2">&quot;</span><span class="nv">$4</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="nv">$5</span><span class="s2">&quot;</span> <span class="p">|</span> samtools sort &gt; <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;indexing aln.bam&quot;</span>
nice samtools index <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;indexing allcontigs.fasta&quot;</span>
nice samtools faidx <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;correcting with Pilon&quot;</span>
nice pilon --genome <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/allcontigs.fasta&quot;</span> --frags <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/aln.bam&quot;</span> --output <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/pilon1&quot;</span> --fix all --mindepth <span class="m">0</span>.5 --changes --verbose --threads <span class="nv">$CPUS</span>
<span class="nb">echo</span>
<span class="nb">echo</span> <span class="s2">&quot;Pilon finished&quot;</span>
<span class="nb">echo</span>

<span class="c1">#count how many changes were made</span>
<span class="nv">pilon_changes</span><span class="o">=</span><span class="k">$(</span>wc -l &lt; <span class="s2">&quot;</span><span class="nv">$1</span><span class="s2">/pilon1.changes&quot;</span><span class="k">)</span>
<span class="nb">echo</span> <span class="s2">&quot;there were </span><span class="nv">$pilon_changes</span><span class="s2"> corrections made&quot;</span>
<span class="nb">echo</span>
<span class="nb">echo</span> <span class="s2">&quot;Assembly complete&quot;</span>
<span class="nb">echo</span> <span class="s2">&quot;The sequence is saved as &quot;</span><span class="nv">$1</span>/pilon1.fasta<span class="s2">&quot;&quot;</span>

<span class="nb">exit</span> <span class="m">1</span>
</pre></div>


<h2 id="run">Run</h2>
<p>To run, type in</p>
<div class="codehilite"><pre><span></span>./myscript.sh outdir <span class="m">1</span>.9m subreads.fastq R1.fastq R2.fastq
</pre></div>


<ul>
<li><strong>outdir</strong> is the name of the output directory</li>
<li><strong>1.9m</strong> is the approximate genome size for the species</li>
<li><strong>subreads.fastq</strong> is the name of the Pacbio reads file</li>
<li><strong>R1.fastq</strong> is the name of the R1 Illumina reads file</li>
<li>
<p><strong>R2.fastq</strong> is the name of the R2 Illumina reads file</p>
</li>
<li>
<p>If the Pacbio and Illumina reads files are not in the current directory, put in the whole path to those files</p>
</li>
<li>
<p>The script should run to the end or may exit early if necessary (e.g. the canu assembly found too many contigs).</p>
</li>
<li>
<p>There are many refinements and additions that could be made to this script, and some of these are noted as comments above.</p>
</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.5e60981f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>