



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/pacbio/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.2.0">
    
    
      
        <title>Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.572ca0f0.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.8c900955.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#assembly-with-pacbio-data-and-smrt-portal" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Home
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy-data/" title="Download and Delete Data in Galaxy" class="md-nav__link">
      Download and Delete Data in Galaxy
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant calling - bacteria" class="md-nav__link">
      Variant calling - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - eukaryotes" class="md-nav__link">
      RNA-seq - eukaryotes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - Bacterial RNA-seq" class="md-nav__link">
      Workflow - Bacterial RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-metagenomics/" title="Workflow - Metagenomics" class="md-nav__link">
      Workflow - Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../help/" title="Help!" class="md-nav__link">
      Help!
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#start" title="Start" class="md-nav__link">
    Start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input" title="Input" class="md-nav__link">
    Input
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" title="Assembly" class="md-nav__link">
    Assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hgap-process-overview" title="HGAP process overview" class="md-nav__link">
    HGAP process overview
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#set-up-job" title="Set up job" class="md-nav__link">
    Set up job
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs-and-outputs" title="Inputs and Outputs" class="md-nav__link">
    Inputs and Outputs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" title="Results" class="md-nav__link">
    Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output" title="Output" class="md-nav__link">
    Output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#links-to-more-information" title="Links to more information" class="md-nav__link">
    Links to more information
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/pacbio/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <p><br></p>
<h1 id="assembly-with-pacbio-data-and-smrt-portal">Assembly with PacBio data and SMRT Portal</h1>
<p>Keywords: de novo assembly, PacBio, PacificBiosciences, HGAP, SMRT Portal, Microbial Genomics Virtual Laboratory</p>
<p>This tutorial will show you how to assemble a bacterial genome <em>de novo</em>, using the PacBio SMRT Portal on the mGVL. We will use an analysis pipeline called HGAP, the Hierarchical Genome Assembly Process.</p>
<!-- FIXMEs:

- will SMRT portal be available on all training GVLs
- will students use existing SMRT portal registrations, or will each set up their own
- PacBio data training data to be loaded into SMRT portal(s) -->

<!-- [How to get data => GVL] -->

<h2 id="start">Start</h2>
<ul>
<li>Open your mGVL dashboard.</li>
<li>You should see SMRT Portal as one of the instance services on your GVL dashboard.</li>
<li>Open up the SMRT portal web link (to the right) and register/log on.</li>
</ul>
<h2 id="input">Input</h2>
<!-- We will use a publicly available data-set of PacBio reads from the bacteria *E. coli* (reference link below).
-->

<p>We will use a dataset from a <em>Streptococcus pyogenes</em> bacteria.</p>
<p>If this has already been loaded onto SMRT portal (e.g. for use during a workshop), proceed to the next step (&ldquo;Assembly&rdquo;).</p>
<p>Otherwise:</p>
<ul>
<li>Load the PacBio data (your own, or the training dataset) onto your GVL.</li>
<li>In the SMRT Portal, go to <ss>Design Job</ss>, the top left tab.</li>
<li>Go to <ss>Import and Manage</ss>.
<img alt="smrt portal screenshot" src="images/image03.png" /></li>
<li>Click <ss>Import SMRT cells</ss>.
<img alt="smrt portal screenshot" src="images/image04.png" /></li>
<li>
<p>Work out where you put the data on your GVL, and make sure the file path is showing.</p>
<ul>
<li>If not, click <ss>Add</ss> and enter the file path to the data.</li>
<li>A SMRT cell is the collection of data from a particular cell in the machine. It includes .bax.h5 files.</li>
</ul>
</li>
<li>
<p>Click on the file path and then <ss>Scan</ss> to check for new data.</p>
</li>
</ul>
<h2 id="assembly">Assembly</h2>
<h3 id="hgap-process-overview">HGAP process overview</h3>
<p>We will use the Hierarchical Genome Assembly Process (HGAP). This flowchart shows the steps in the process:</p>
<p><img alt="flowchart of HGAP process" src="images/flowchart.png" /></p>
<h3 id="set-up-job">Set up job</h3>
<ul>
<li>In the SMRT Portal, go to the top left tab, <ss>Design Job</ss>.</li>
<li>Go to <ss>Create New</ss>.</li>
<li>An <ss>Analysis</ss> window should appear. Tick all the boxes, then <ss>Next</ss>.</li>
<li>Under <ss>Job Name</ss> enter a name.</li>
<li>To the right, under <ss>Groups</ss> choose <em>all</em>.</li>
<li>Under <ss>Protocols</ss> choose <ss>RS_HGAP_Assembly.3</ss>.</li>
<li>There is an ellipsis underneath <ss>Protocols</ss> - click on the ellipsis.</li>
</ul>
<p><img alt="smrt portal screenshot" src="images/setup.png" /></p>
<p>This brings up the settings. Click on <ss>Assembly</ss>.</p>
<ul>
<li>
<p>For <ss>Compute Minimum Seed Read Length</ss>: ensure box is ticked
<!--
Sarah Baines:
("compute minimum seed read length"). While this protocol works ok, I have found that the longer the average seed read length (larger library cuts), the estimation protocol can get it really wrong, and frequently if the cell has low output this will cause Celera to fail very early on. Performing this estimation manually can be difficult (as you know!) but it may be worth adding a comment or something.
--></p>
</li>
<li>
<p>For <ss>Number of Seed Read Chunks</ss>: enter <em>12</em></p>
</li>
<li>Change the <ss>Genome Size</ss> to an approximately correct size for the species. For <em>S. pyogenes</em>, enter 1800000.</li>
<li>For <ss>Target Coverage</ss>: enter <em>10</em></li>
<li>For <ss>Overlapper Error Rate</ss>: enter <em>0.04</em></li>
<li>Leave all other settings as they are.</li>
<li>Click <ss>Apply</ss></li>
</ul>
<p>Your protocol window should look like this:</p>
<p><img alt="smrt portal screenshot" src="images/assembly_options.png" /></p>
<ul>
<li>
<p>Click <ss>Ok</ss>.  </p>
</li>
<li>
<p>In the <ss>SMRT Cells Available</ss> window, select the file to be used. Click on the arrow to transfer these files to the SMRT Cells in Job window.</p>
</li>
<li>You can drag the column widths of the &ldquo;Url&rdquo; column so that you can see the URLs of the file paths better.</li>
</ul>
<p><img alt="smrt portal screenshot" src="images/smrt3.png" /></p>
<ul>
<li>Click <ss>Save</ss> (bottom right hand side).</li>
<li>Next to <ss>Save</ss>, click <ss>Start</ss>.</li>
<li>The <ss>Monitor Jobs</ss> window should open.<ul>
<li>As each step proceeds, new items will appear under the <ss>Reports</ss> and <ss>Data</ss> tabs on the left.</li>
</ul>
</li>
</ul>
<p><img alt="smrt portal screenshot" src="images/smrt6.png" /></p>
<h3 id="inputs-and-outputs">Inputs and Outputs</h3>
<p>The connections between the names of assembly stages and outputs is not always clear. This flowchart shows how each stage of the HGAP process corresponds to protocol window names and outputs:</p>
<p><img alt="inputs and outputs" src="images/inputs_outputs.png" />
<!-- upload a better res image? --></p>
<h2 id="results">Results</h2>
<p>If the job is still running, click on the centre tab <ss>Monitor Jobs</ss>. Otherwise, click on the top right tab, <ss>View Data</ss>.</p>
<ul>
<li>Double click on the job name to open its reports.</li>
<li>Click on different <ss>Reports</ss> in the left hand panel.</li>
</ul>
<p>Things to look at:</p>
<p><strong>General: Filtering (polymerase reads)</strong></p>
<ul>
<li>number of reads post-filter</li>
<li>read length (=average)</li>
</ul>
<p><img alt="filtering results" src="images/filtering_results.png" /></p>
<p><strong>General: Subread Filtering (subreads)</strong></p>
<ul>
<li>number of reads post-filter</li>
<li>read length (average)</li>
</ul>
<p><img alt="subreads results" src="images/subreads_results.png" /></p>
<p><strong>Assembly: Pre-Assembly (pre-assembled reads)</strong></p>
<ul>
<li>length cutoff (the computed minimum seed read length)</li>
<li>read length (average)</li>
</ul>
<p><img alt="preassembly results" src="images/preassembly_results.png" /></p>
<p><strong>Assembly: Corrections</strong></p>
<p>Consensus calling results:</p>
<ul>
<li>Consensus concordance should be &gt; 99%.</li>
</ul>
<p>Graph: corrections across reference:</p>
<ul>
<li>With the first run of polishing, we expect a lot of corrections but they should be randomly distributed.</li>
</ul>
<p><img alt="corrections results" src="images/corrections_results.png" /></p>
<p><em>Note: only unitigs 0 and 1 shown.</em></p>
<p><strong>Assembly: Top Corrections</strong></p>
<p>This is a list of all the corrections made.</p>
<p><img alt="top corrections results" src="images/topcorrections_results.png" /></p>
<p><em>Note: only first 15 shown.</em></p>
<!-- - If more than two corrections (with confidence > 50), repeat polishing (see next section "Further polishing"). -->

<p><strong>Resequencing: Coverage</strong></p>
<p>Coverage across reference:</p>
<ul>
<li>discard contigs &lt;20X coverage</li>
<li>others should have fairly consistent coverage.</li>
<li>spikes could be collapsed repeats.</li>
<li>valleys could be mis-assembly - e.g. draft assembly was incorrect and so remapped reads didn&rsquo;t support this part of the assembly.</li>
</ul>
<p><img alt="resequencing coverage results" src="images/reseq_coverage_results.png" /></p>
<p><em>Note: only unitigs 0 and 1 shown.</em></p>
<p>Graph: Depth of Coverage:</p>
<p>The is the <em>number</em> of reference regions vs. coverage. <!-- Should be roughly normal? --></p>
<p><img alt="depth coverage results" src="images/depth_coverage_results.png" /></p>
<p><strong>Assembly: Polished Assembly</strong></p>
<ul>
<li>Number of contigs</li>
<li>Max contig length</li>
<li>Graph: confidence vs depth. Multi-copy plasmids may have higher coverage.</li>
</ul>
<p><img alt="polished assembly results" src="images/polished_assembly_results.png" /></p>
<!--
Sarah Baines: I have found with our data that there is no point in repeating quiver more than once after HGAP, and especially if we are going to polish the assembly with short read data later. A third run has never made a significant improvement to any of the genomes I have worked with, and after a few runs quiver starts to cycle - where is makes a correction then corrects it back to the original call in the subsequent run and so on.

- therefore I have hidden this section for now - AS.

## Further Polishing

During polishing, raw reads are used to correct the assembly.
During HGAP, the assembly was polished once but may need further corrections.

- From the previous step, Go to <ss>Data &rarr; Assembly &rarr; Polished Assembly</ss> and download the FASTA file by clicking on it.
    - Unzip the .gz file
- Go to <ss>Design Job &rarr; Import and Manage</ss> and click <ss>New</ss> on the bottom right hand side. Then, select that FASTA assembly file to upload.
    - creates a new reference.
- Go to <ss>Design Job &rarr; Create New</ss>
    - choose reference-based
    - Select protocol: RS_Resequencing.1
    - Leave all settings.
    - Select your reference from the drop down menu.
    - Click <ss>Save</ss> and <ss>Start</ss>.
- Examine the output assembly and repeat if necessary (e.g. if > 2 corrections with >50 confidence).
-->

<h2 id="output">Output</h2>
<p>The polished assembly as a FASTA file.</p>
<p><img alt="output fasta file" src="images/output_fasta.png" /></p>
<ul>
<li>download to local computer; or</li>
<li>open file in (GVL) Galaxy; or</li>
<li>open file in GVL command line and perform further analysis.</li>
</ul>
<!-- ## Correct with short reads

If you have Illumina reads for the same sample, the assembly can be further polished.

- Download the FASTA assembly to your computer.
- Open Galaxy
- Upload the FASTA assembly
- Upload the Illumina reads to galaxy
- change datatype to fastqsanger
ngs mapping - map with bwa mem
-  use genome from history - the smrt portal fasta file
R1 and R2 files
execute
refresh

- Polish with Pilon.

pilon
use a genome from history : the smrt portal fasta file
input bam file - from step above
variant calling mode - no
create changes file - yes
advanced options:
paired end : yes
fix - select all
mindepth 0.5



output = polished assembly
-->

<h2 id="next">Next</h2>
<p>Further options:</p>
<ul>
<li>
<p>Circularise the contigs; e.g. with Circlator.</p>
</li>
<li>
<p>Polish the assembly using Illumina data; e.g. with Pilon.</p>
</li>
<li>
<p>Annotate the polished assembly; e.g. with Prokka.</p>
</li>
</ul>
<h2 id="links-to-more-information">Links to more information</h2>
<p><a href="https://github.com/PacificBiosciences/DevNet/wiki/E.-coli-Bacterial-Assembly">PacBio <em>E. coli</em> data set</a></p>
<p><a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP">HGAP overview</a></p>
<p><a href="http://files.pacb.com/software/smrtanalysis/2.3.0/doc/smrtportal/help/Webhelp/SMRT_Portal.htm">A full ist of reports and terminology</a></p>
<p><a href="http://www.pacb.com/training/BacterialAssemblyandEpigeneticAnalysis/story.html">Video overview of HGAP on SMRT portal</a></p>
<p><a href="http://files.pacb.com/Training/IntroductiontoSMRTbellTemplatePreparation/story_content/external_files/Introduction%20to%20SMRTbell%E2%84%A2%20Template%20Preparation.pdf">More about the SMRT bell template</a></p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b41f3d20.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>