



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/cmdline_assembly/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.6">
    
    
      
        <title>Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.451f80e5.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#pacbio-reads-assembly-with-command-line-tools" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Home
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy-data/" title="Download and Delete Data in Galaxy" class="md-nav__link">
      Download and Delete Data in Galaxy
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant calling - bacteria" class="md-nav__link">
      Variant calling - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - eukaryotes" class="md-nav__link">
      RNA-seq - eukaryotes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - Bacterial RNA-seq" class="md-nav__link">
      Workflow - Bacterial RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-metagenomics/" title="Workflow - Metagenomics" class="md-nav__link">
      Workflow - Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#resources" title="Resources" class="md-nav__link">
    Resources
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pacbio-files" title="PacBio files" class="md-nav__link">
    PacBio files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#illumina-files" title="Illumina files" class="md-nav__link">
    Illumina files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sample-information" title="Sample information" class="md-nav__link">
    Sample information
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble" title="Assemble" class="md-nav__link">
    Assemble
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#check-the-output" title="Check the output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#change-canu-parameters-if-required" title="Change Canu parameters if required" class="md-nav__link">
    Change Canu parameters if required
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#trim-and-circularise" title="Trim and circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output_1" title="Check the output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#options" title="Options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#find-smaller-plasmids" title="Find smaller plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plasmid-contig-orientation" title="Plasmid contig orientation" class="md-nav__link">
    Plasmid contig orientation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct" title="Correct" class="md-nav__link">
    Correct
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-output" title="Genome output" class="md-nav__link">
    Genome output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advanced-analysis" title="Advanced analysis" class="md-nav__link">
    Advanced analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assemble_1" title="Assemble" class="md-nav__link">
    Assemble
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-and-circularise_1" title="Trim and circularise" class="md-nav__link">
    Trim and circularise
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#find-smaller-plasmids_1" title="Find smaller plasmids" class="md-nav__link">
    Find smaller plasmids
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#correct_1" title="Correct" class="md-nav__link">
    Correct
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/cmdline_assembly/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <p><br></p>
<h1 id="pacbio-reads-assembly-with-command-line-tools">Pacbio reads: assembly with command line tools</h1>
<p><em>Keywords: de novo assembly, PacBio, PacificBiosciences, Illumina, command line, Canu, Circlator, BWA, Spades, Pilon, Microbial Genomics Virtual Laboratory</em></p>
<p>This tutorial demonstrates how to use long Pacbio sequence reads to assemble a bacterial genome, including correcting the assembly with short Illumina reads.</p>
<p>Note: this tutorial uses canu v1.5. </p>
<h2 id="resources">Resources</h2>
<p>Tools (and versions) used in this tutorial include:</p>
<ul>
<li>canu 1.5 </li>
<li>infoseq and sizeseq (part of EMBOSS) 6.6.0.0</li>
<li>circlator 1.5.1 [recently updated]</li>
<li>bwa 0.7.15</li>
<li>samtools 1.3.1</li>
<li>spades 3.10.1</li>
<li>makeblastdb and blastn (part of blast) 2.4.0+</li>
<li>pilon 1.20</li>
</ul>
<!-- check these are the correct version - updates? -->

<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial, be able to:</p>
<ol>
<li>Assemble and circularise a bacterial genome from PacBio sequence data.</li>
<li>Recover small plasmids missed by long read sequencing, using Illumina data</li>
<li>Explore the effect of polishing assembled sequences with a different data set.</li>
</ol>
<h2 id="overview">Overview</h2>
<p>Simplified version of workflow:</p>
<p><img alt="workflow" src="images/flowchart.png" /></p>
<h2 id="get-data">Get data</h2>
<p>The files we need are:</p>
<ul>
<li><fn>pacbio.fastq.gz</fn> : the PacBio reads</li>
<li><fn>illumina_R1.fastq.gz</fn>: the Illumina forward reads</li>
<li><fn>illumina_R2.fastq.gz</fn>: the Illumina reverse reads</li>
</ul>
<p>If you already have the files, skip forward to next section, <a href="#assemble">Assemble</a>.</p>
<p>Otherwise, this section has information about how to find and move the files:</p>
<h3 id="pacbio-files">PacBio files</h3>
<ul>
<li>Open the command line. <!-- own machine, mGVL or BPA VM --></li>
<li>Navigate to or create the directory in which you want to work.</li>
<li>If the files are already on your server, you can symlink by using</li>
</ul>
<div class="codehilite"><pre><span></span>ln -s real_file_path [e.g. data/sample_name/pacbio1.fastq.gz] chosen_symlink_name [e.g. pacbio1.fastq.gz]
</pre></div>


<ul>
<li>
<p>Alternatively, obtain the input files from elsewhere, e.g. from the BPA portal. (You will need a password.)</p>
</li>
<li>
<p>Pacbio files are often stored in the format:</p>
<ul>
<li><fn>Sample_name/Cell_name/Analysis_Results/long_file_name_1.fastq.gz</fn></li>
</ul>
</li>
<li>
<p>We will use the <fn>longfilename.subreads.fastq.gz</fn> files.</p>
</li>
<li>
<p>The reads are usually split into three separate files because they are so large.</p>
</li>
<li>
<p>Right click on the first <fn>subreads.fastq.gz</fn> file and &ldquo;copy link address&rdquo;.</p>
</li>
<li>
<p>In the command line, type:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span>wget --user username --password password [paste link URL for file]
</pre></div>


<ul>
<li>Repeat for the other two <fn>subreads.fastq.gz</fn> files.</li>
<li>Join the files:</li>
</ul>
<div class="codehilite"><pre><span></span>cat pacbio*.fastq.gz &gt; pacbio.fastq.gz
</pre></div>


<ul>
<li>If the files are not gzipped, type:</li>
</ul>
<div class="codehilite"><pre><span></span>cat pacbio*.fastq | gzip &gt; pacbio.fastq.gz
</pre></div>


<h3 id="illumina-files">Illumina files</h3>
<ul>
<li>We will also use 2 x Illumina (Miseq) fastq.gz files.</li>
<li>These are the <fn>R1.fastq.gz</fn> and <fn>R2.fastq.gz</fn> files.</li>
<li>Symlink or &ldquo;wget&rdquo; these files as described above for PacBio files.</li>
<li>Shorten the name of each of these files:</li>
</ul>
<div class="codehilite"><pre><span></span>mv longfilename_R1.fastq.gz illumina_R1.fastq.gz
mv longfilename_R2.fastq.gz illumina_R2.fastq.gz
</pre></div>


<!--
Find information about read lengths: fq subreads.fastq.gz
- Look at the average and maximum lengths.
-->

<h3 id="sample-information">Sample information</h3>
<p>The sample used in this tutorial is a gram-positive bacteria called <em>Staphylococcus aureus</em> (sample number 25747). This particular sample is from a strain that is resistant to the antibiotic methicillin (a type of penicillin). It is also called MRSA: methicillin-resistant <em>Staphylococcus aureus</em>. It was isolated from (human) blood and caused bacteraemia, an infection of the bloodstream.</p>
<h2 id="assemble">Assemble<a name="assemble"></a></h2>
<ul>
<li>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>.</li>
<li>Run Canu with these commands:</li>
</ul>
<div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir genomeSize=2.8m -pacbio-raw pacbio.fastq.gz
</pre></div>


<ul>
<li>the first <code>canu</code> tells the program to run</li>
<li><code>-p canu</code> names prefix for output files (&ldquo;canu&rdquo;)</li>
<li><code>-d canu_outdir</code> names output directory (&ldquo;canu_outdir&rdquo;)</li>
<li>
<p><code>genomeSize</code> only has to be approximate.</p>
<ul>
<li>e.g. <em>Staphylococcus aureus</em>, 2.8m</li>
<li>e.g. <em>Streptococcus pyogenes</em>, 1.8m</li>
</ul>
</li>
<li>
<p>Canu will correct, trim and assemble the reads.</p>
</li>
<li>Various output will be displayed on the screen.</li>
</ul>
<h3 id="check-the-output">Check the output</h3>
<p>Move into <fn>canu_outdir</fn> and <code>ls</code> to see the output files.</p>
<ul>
<li>The <fn>canu.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>canu.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>canu.correctedReads.fasta.gz</fn> are the corrected Pacbio reads that were used in the assembly.</li>
<li>The <fn>canu.file.gfa</fn> is the graph of the assembly.</li>
<li>Display summary information about the contigs: (<code>infoseq</code> is a tool from <a href="http://emboss.sourceforge.net/index.html">EMBOSS</a>)</li>
</ul>
<div class="codehilite"><pre><span></span>infoseq canu.contigs.fasta
</pre></div>


<ul>
<li>This will show the contigs found by Canu. e.g.,</li>
</ul>
<div class="codehilite"><pre><span></span>    - tig00000001   2851805
</pre></div>


<p>This looks like a chromosome of approximately 2.8 million bases.</p>
<p>This matches what we would expect for this sample. For other data, Canu may not be able to join all the reads into one contig, so there may be several contigs in the output. Also, the sample may contain some plasmids and these may be found full or partially by Canu as additional contigs.  </p>
<h3 id="change-canu-parameters-if-required">Change Canu parameters if required</h3>
<p>If the assembly is poor with many contigs, re-run Canu with extra sensitivity parameters; e.g.</p>
<div class="codehilite"><pre><span></span>canu -p prefix -d outdir corMhapSensitivity=high corMinCoverage=0 genomeSize=2.8m -pacbio-raw pacbio.fastq.gz
</pre></div>


<h3 id="questions">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do long- and short-read assembly methods differ?</p>
<details><summary>Answer</summary><p>short reads: De Bruijn graphs; long reads: a move back towards simpler overlap-layout-consensus methods.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where can we find out the what the approximate genome size should be for the species being assembled?</p>
<details><summary>Answer</summary><p>NCBI Genomes - enter species name - click on Genome Assembly and Annotation report - sort table by clicking on the column header Size (Mb) - look at range of sizes in this column.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>In the assembly output, what are the unassembled reads? Why are they there?</p>
<details><summary>Answer</summary></details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are the corrected reads? How did canu correct the reads?</p>
<details><summary>Answer</summary></details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Where could you view the output .gfa and what would it show?</p>
<details><summary>Answer</summary></details>
</div>
<h2 id="trim-and-circularise">Trim and circularise</h2>
<h3 id="run-circlator">Run Circlator</h3>
<p>Circlator identifies and trims overhangs (on chromosomes and plasmids) and orients the start position at an appropriate gene (e.g. dnaA). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>Overhangs are shown in blue:</p>
<p><img alt="circlator" src="images/circlator_diagram.png" />
<em>Adapted from Figure 1. Hunt et al. Genome Biology 2015</em></p>
<p>Move back into your main analysis folder.</p>
<p>Run Circlator:</p>
<div class="codehilite"><pre><span></span>circlator all --threads 8 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir
</pre></div>


<ul>
<li><code>--threads</code> is the number of cores: change this to an appropriate number</li>
<li><code>--verbose</code> prints progress information to the screen</li>
<li><code>canu_outdir/canu.contigs.fasta</code> is the file path to the input Canu assembly</li>
<li><code>canu_outdir/canu.correctedReads.fasta.gz</code> is the file path to the corrected Pacbio reads - note, fastA not fastQ</li>
<li><code>circlator_outdir</code> is the name of the output directory.</li>
</ul>
<p>Some output will print to screen. When finished, it should say &ldquo;Circularized x of x contig(s)&rdquo;.</p>
<h3 id="check-the-output_1">Check the output</h3>
<p>Move into the <fn>circlator_outdir</fn> directory and <code>ls</code> to list files.</p>
<p><em>Were the contigs circularised?</em> :</p>
<div class="codehilite"><pre><span></span>less 04.merge.circularise.log
</pre></div>


<ul>
<li>Yes, the contig was circularised (last column).</li>
<li>Type &ldquo;q&rdquo; to exit.</li>
</ul>
<p><em>Where were the contigs oriented (which gene)?</em> :</p>
<div class="codehilite"><pre><span></span>less 06.fixstart.log
</pre></div>


<ul>
<li>Look in the &ldquo;gene_name&rdquo; column.</li>
<li>The contig has been oriented at tr|A0A090N2A8|A0A090N2A8_STAAU, which is another name for dnaA. <!-- (search swissprot - uniprot.org) --> This is typically used as the start of bacterial chromosome sequences.</li>
</ul>
<p><em>What are the trimmed contig sizes?</em> :</p>
<div class="codehilite"><pre><span></span>infoseq 06.fixstart.fasta
</pre></div>


<ul>
<li>tig00000001 2823331 (28564 bases trimmed)</li>
</ul>
<p>This trimmed part is the overlap.</p>
<p><em>Re-name the contigs file</em>:</p>
<ul>
<li>The trimmed contigs are in the file called <fn>06.fixstart.fasta</fn>.</li>
<li>Re-name it <fn>contig1.fasta</fn>:</li>
</ul>
<div class="codehilite"><pre><span></span>mv 06.fixstart.fasta contig1.fasta
</pre></div>


<p>Open this file in a text editor (e.g. nano: <code>nano contig1.fasta</code>) and change the header to &ldquo;&gt;chromosome&rdquo;.</p>
<p>Move the file back into the main folder (<code>mv contig1.fasta ../</code>).</p>
<h3 id="options">Options</h3>
<p>If all the contigs have not circularised with Circlator, an option is to change the <code>--b2r_length_cutoff</code> setting to approximately 2X the average read depth.</p>
<h3 id="questions_1">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Were all the contigs circularised? Why/why not?</p>
<details><summary>Answer</summary></details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Circlator can set the start of the sequence at a particular gene. Which gene does it use? Is this appropriate for all contigs?</p>
<details><summary>Answer</summary><p>Uses dnaA for the chromosomal contig. For other contigs, uses a centrally-located gene. However, ideally, plasmids would be oriented on a gene such as repA. It is possible to provide a file to Circlator to do this.</p>
</details>
</div>
<h2 id="find-smaller-plasmids">Find smaller plasmids</h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the Canu+Circlator output of a trimmed assembly contig.</li>
<li>Map all the Illumina reads against this Pacbio-assembled contig.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim.</li>
</ol>
<h3 id="align-illumina-reads-to-the-pacbio-contig">Align Illumina reads to the PacBio contig</h3>
<ul>
<li>Index the contigs file:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa index contig1.fasta
</pre></div>


<ul>
<li>Align Illumina reads using using bwa mem:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa mem -t 8 contig1.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln.bam
</pre></div>


<ul>
<li><code>bwa mem</code> is the alignment tool</li>
<li><code>-t 8</code> is the number of cores: choose an appropriate number</li>
<li><code>contig1.fasta</code> is the input assembly file</li>
<li><code>illumina_R1.fastq.gz illumina_R2.fastq.gz</code> are the Illumina reads</li>
<li><code>| samtools sort</code> pipes the output to samtools to sort</li>
<li><code>&gt; aln.bam</code> sends the alignment to the file <fn>aln.bam</fn></li>
</ul>
<h3 id="extract-unmapped-illumina-reads">Extract unmapped Illumina reads</h3>
<ul>
<li>Index the alignment file:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools index aln.bam
</pre></div>


<ul>
<li>Extract the fastq files from the bam alignment - those reads that were unmapped to the Pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</pre></div>


<ul>
<li><code>fastq</code> is a command that coverts a <fn>.bam</fn> file into fastq format</li>
<li><code>-f 4</code> : only output unmapped reads</li>
<li><code>-1</code> : put R1 reads into a file called <fn>unmapped.R1.fastq</fn></li>
<li><code>-2</code> : put R2 reads into a file called <fn>unmapped.R2.fastq</fn></li>
<li><code>-s</code> : put singleton reads into a file called <fn>unmapped.RS.fastq</fn></li>
<li><code>aln.bam</code> : input alignment file</li>
</ul>
<p>We now have three files of the unampped reads: <fn> unmapped.R1.fastq</fn>, <fn> unmapped.R2.fastq</fn>, <fn> unmapped.RS.fastq</fn>.</p>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads</h3>
<ul>
<li>Assemble with Spades:</li>
</ul>
<div class="codehilite"><pre><span></span>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly
</pre></div>


<ul>
<li><code>-1</code> is input file forward</li>
<li><code>-2</code> is input file reverse</li>
<li><code>-s</code> is unpaired</li>
<li><code>--careful</code> minimizes mismatches and short indels</li>
<li><code>--cov-cutoff auto</code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code>-o</code> is the output directory</li>
</ul>
<p>Move into the output directory (<fn>spades_assembly</fn>) and look at the contigs:</p>
<div class="codehilite"><pre><span></span>infoseq contigs.fasta
</pre></div>


<ul>
<li>78 contigs were assembled, with the max length of 2250 (the first contig).  </li>
<li>All other nodes are &lt; 650kb so we will disregard as they are unlikely to be plasmids.</li>
<li>Type &ldquo;q&rdquo; to exit.</li>
<li>We will extract the first sequence (NODE_1):</li>
</ul>
<div class="codehilite"><pre><span></span>samtools faidx contigs.fasta
</pre></div>


<div class="codehilite"><pre><span></span>samtools faidx contigs.fasta NODE_1_length_2550_cov_496.613 &gt; contig2.fasta
</pre></div>


<ul>
<li>This is now saved as <fn>contig2.fasta</fn></li>
<li>Open in nano and change header to &ldquo;&gt;plasmid&rdquo;.</li>
</ul>
<h3 id="trim-the-plasmid">Trim the plasmid</h3>
<p>To trim any overhang on this plasmid, we will blast the start of contig2 against itself.</p>
<ul>
<li>Take the start of the contig:</li>
</ul>
<div class="codehilite"><pre><span></span>head -n 10 contig2.fasta &gt; contig2.fa.head
</pre></div>


<ul>
<li>We want to see if it matches the end (overhang).</li>
<li>Format the assembly file for blast:</li>
</ul>
<div class="codehilite"><pre><span></span>makeblastdb -in contig2.fasta -dbtype nucl
</pre></div>


<ul>
<li>Blast the start of the assembly (.head file) against all of the assembly:</li>
</ul>
<div class="codehilite"><pre><span></span>blastn -query contig2.fa.head -db contig2.fasta -evalue 1e-3 -dust no -out contig2.bls
</pre></div>


<ul>
<li>Look at <fn>contig2.bls</fn> to see hits:</li>
</ul>
<div class="codehilite"><pre><span></span>less contig2.bls
</pre></div>


<ul>
<li>The first hit is at start, as expected.</li>
<li>The second hit is at 2474 all the way to the end - 2550.</li>
<li>This is the overhang.</li>
<li>Trim to position 2473.</li>
<li>Index the plasmid.fa file:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta
</pre></div>


<ul>
<li>Trim:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools faidx contig2.fasta plasmid:1-2473 &gt; plasmid.fa.trimmed
</pre></div>


<ul>
<li>
<p><code>plasmid</code> is the name of the contig, and we want the sequence from 1-2473.</p>
</li>
<li>
<p>Open this file in nano (<code>nano plasmid.fa.trimmed</code>) and change the header to &ldquo;&gt;plasmid&rdquo;, save.</p>
</li>
<li>We now have a trimmed plasmid.</li>
<li>Move file back into main folder:</li>
</ul>
<div class="codehilite"><pre><span></span>cp plasmid.fa.trimmed ../
</pre></div>


<ul>
<li>Move into the main folder.</li>
</ul>
<h3 id="plasmid-contig-orientation">Plasmid contig orientation</h3>
<p>The bacterial chromosome was oriented at the gene dnaA. Plasmids are often oriented at the replication gene, but this is highly variable and there is no established convention. Here we will orient the plasmid at a gene found by Prodigal, in Circlator:</p>
<div class="codehilite"><pre><span></span>circlator fixstart plasmid.fa.trimmed plasmid_fixstart
</pre></div>


<ul>
<li><code>fixstart</code> is an option in Circlator just to orient a sequence.</li>
<li><code>plasmid.fa.trimmed</code> is our small plasmid.</li>
<li><code>plasmid_fixstart</code> is the prefix for the output files.</li>
</ul>
<p>View the output:</p>
<div class="codehilite"><pre><span></span>less plasmid_fixstart.log
</pre></div>


<ul>
<li>The plasmid has been oriented at a gene predicted by Prodigal, and the break-point is at position 1200.</li>
<li>Change the file name:</li>
</ul>
<div class="codehilite"><pre><span></span>cp plasmid_fixstart.fasta contig2.fasta
</pre></div>


<!-- note: annotated with prokka. plasmid only has two proteins. ermC, and a hypothetical protein. protein blast genbank: matches a staph replication and maintenance protein. -->

<h3 id="collect-contigs">Collect contigs</h3>
<div class="codehilite"><pre><span></span>cat contig1.fasta contig2.fasta &gt; genome.fasta
</pre></div>


<ul>
<li>See the contigs and sizes:</li>
</ul>
<div class="codehilite"><pre><span></span>infoseq genome.fasta
</pre></div>


<ul>
<li>chromosome: 2823331</li>
<li>plasmid: 2473</li>
</ul>
<h3 id="questions_2">Questions</h3>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why is this section so complicated?</p>
<details><summary>Answer</summary><p>Finding small plasmids is difficult for many reasons! This paper has a nice summary: On the (im)possibility to reconstruct plasmids from whole genome short-read sequencing data. doi: https://doi.org/10.1101/086744</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can PacBio sequencing miss small plasmids?</p>
<details><summary>Answer</summary><p>Library prep size selection</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>We extract unmapped Illumina reads and assemble these to find small plasmids. What could they be missing?</p>
<details><summary>Answer</summary><p>Repeats that have mapped to the PacBio assembly.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How do you find a plasmid in a Bandage graph?</p>
<details><summary>Answer</summary><p>It is probably circular, matches the size of a known plasmid, and has a rep gene.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Are there easier ways to find plasmids?</p>
<details><summary>Answer</summary><p>Possibly. One option is the program called Unicycler which may automate many of these steps. https://github.com/rrwick/Unicycler</p>
</details>
</div>
<h2 id="correct">Correct</h2>
<p>We will correct the Pacbio assembly with Illumina reads.</p>
<!--
First, we will change some of the mGVL settings so that we can use a worker node with more CPUs.

Type in


wzxhzdk:34


(To later exit out of this worker node, if you want to, type in "exit").
--->

<h3 id="make-an-alignment-file">Make an alignment file</h3>
<ul>
<li>Align the Illumina reads (R1 and R2) to the draft PacBio assembly, e.g. <fn>genome.fasta</fn>:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa index genome.fasta
bwa mem -t 32 genome.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln.bam
</pre></div>


<ul>
<li>
<p><code>-t</code> is the number of cores: set this to an appropriate number. (To find out how many you have, <code>grep -c processor /proc/cpuinfo</code>).</p>
</li>
<li>
<p>Index the files:</p>
</li>
</ul>
<div class="codehilite"><pre><span></span>samtools index aln.bam
samtools faidx genome.fasta
</pre></div>


<ul>
<li>Now we have an alignment file to use in Pilon: <fn>aln.bam</fn></li>
</ul>
<!--
Look at how the illumina reads are aligned:

wzxhzdk:37

note: **contig1** is the name of the contig to view; e.g. tig00000000.

[explain settings in tview - how to scroll, find]
-->

<h3 id="run-pilon">Run Pilon</h3>
<ul>
<li>Run:</li>
</ul>
<div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --verbose --threads 32
</pre></div>


<ul>
<li><code>--genome</code> is the name of the input assembly to be corrected</li>
<li><code>--frags</code> is the alignment of the reads against the assembly</li>
<li><code>--output</code> is the name of the output prefix</li>
<li><code>--fix</code> is an option for types of corrections</li>
<li><code>--mindepth</code> gives a minimum read depth to use</li>
<li><code>--changes</code> produces an output file of the changes made</li>
<li><code>--verbose</code> prints information to the screen during the run</li>
<li><code>--threads</code> : set this to an appropriate number</li>
</ul>
<p>Look at the changes file:</p>
<div class="codehilite"><pre><span></span>less pilon1.changes
</pre></div>


<p><em>Example:</em></p>
<p><img alt="pilon" src="images/pilon.png" /></p>
<p>Look at the details of the fasta file:</p>
<div class="codehilite"><pre><span></span>infoseq pilon1.fasta
</pre></div>


<ul>
<li>chromosome - 2823340 (net +9 bases)</li>
<li>plasmid - 2473 (no change)</li>
</ul>
<p><strong>Option:</strong></p>
<p>If there are many changes, run Pilon again, using the <fn>pilon1.fasta</fn> file as the input assembly, and the Illumina reads to correct.</p>
<h3 id="genome-output">Genome output</h3>
<ul>
<li>Change the file name:</li>
</ul>
<div class="codehilite"><pre><span></span>cp pilon1.fasta assembly.fasta
</pre></div>


<ul>
<li>We now have the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing a chromosome and a small plasmid.  </li>
</ul>
<h3 id="questions_3">Questions</h3>
<p>Q:</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why don&rsquo;t we correct earlier in the assembly process?</p>
<details><summary>Answer</summary><p>We need to circularise the contigs and trim overhangs first.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why can we use some reads (Illumina) to correct other reads (PacBio) ?</p>
<details><summary>Answer</summary><p>Illumina reads have higher accuracy.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Could we just use PacBio reads to assemble the genome?</p>
<details><summary>Answer</summary><p>Yes, if accuracy adequate.</p>
</details>
</div>
<h2 id="advanced-analysis">Advanced analysis</h2>
<p>This example shows a more complex analysis where many more steps are involved in the finding the small plasmid. The sample used is <em>Staphylococcus aureus</em> (sample number 25745).</p>
<h3 id="assemble_1">Assemble</h3>
<div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir genomeSize=2.8m -pacbio-raw pacbio.fastq.gz
</pre></div>


<ul>
<li>Output: 2 contigs, likely to be the chromosome (2748030) and a large plasmid (49397).</li>
</ul>
<h3 id="trim-and-circularise_1">Trim and circularise</h3>
<div class="codehilite"><pre><span></span>circlator all --threads 16 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir
</pre></div>


<ul>
<li>Look at the information about circularisation, orientation, and trimmed sizes.</li>
<li>Re-name the file <fn>contigs_1_2.fasta</fn> and move it into the main folder.</li>
</ul>
<h3 id="find-smaller-plasmids_1">Find smaller plasmids</h3>
<ul>
<li>Align Illumina reads to the PacBio assembly:</li>
</ul>
<div class="codehilite"><pre><span></span>bwa index contigs_1_2.fasta
bwa mem -t 8 contigs_1_2.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln.bam
</pre></div>


<div class="codehilite"><pre><span></span>samtools index aln.bam
samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly
</pre></div>


<ul>
<li>Look at the output:</li>
</ul>
<div class="codehilite"><pre><span></span>cd spades_assembly
infoseq contigs.fasta
</pre></div>


<ul>
<li>Extract the first node:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools faidx contigs.fasta
samtools faidx contigs.fasta NODE_1_length_2229_cov_610.298 &gt; contig3.fasta
</pre></div>


<ul>
<li>Open in Nano and change header to &ldquo;plasmid&rdquo;.</li>
<li>Look for overhang by blasting start of plamsid against itself:</li>
</ul>
<div class="codehilite"><pre><span></span>head -n 10 contig3.fasta &gt; contig3.fa.head
makeblastdb -in contig3.fasta -dbtype nucl
blastn -query contig3.fa.head -db contig3.fasta -evalue 1e-3 -dust no -out contig3.bls
less contig3.bls
</pre></div>


<ul>
<li>There is only one hit, to the start of the plasmid. No overhang is found.</li>
<li>Search Genbank for any matching proteins: Copy the sequence</li>
<li>Go to NCBI: <tt>https://blast.ncbi.nlm.nih.gov/Blast.cgi</tt>; choose blastx</li>
<li>Paste the sequence from <fn>contig3.fasta</fn></li>
<li>Choose genetic code = 11</li>
<li>Blast</li>
<li>This hits a replication (plasmid) protein. Hypothesise that   this is a small plasmid; search for the entire sequence within the assembly of all the Illumina reads (next step).</li>
<li>Copy <fn>contig3.fasta</fn> into the main folder.</li>
<li>Assemble all the Illumina reads and produce an assembly graph.</li>
</ul>
<div class="codehilite"><pre><span></span>spades.py -1 illumina_R1.fastq.gz -2 illumina_R2.fastq.gz --careful --cov-cutoff auto -o spades_assembly_all_illumina
</pre></div>


<ul>
<li>Navigate to the output and find the <fn>assembly_graph.fastg</fn>.</li>
<li>Transfer this file to your local computer (e.g. using the file transfer program <a href="https://cyberduck.io/?l=en">Cyberduck</a>).</li>
<li>Examine the assembly in the program <a href="https://rrwick.github.io/Bandage/">Bandage</a>.<ul>
<li>File: Load graph: <fn>assembly_graph.fastg</fn></li>
<li>In the left hand panel, click &ldquo;Draw graph&rdquo;</li>
<li>Your assembly graph may look like this:
<img alt="bandage pic" src="images/illumina_assembly_bandage.png" /></li>
</ul>
</li>
<li>
<p>Blast the small plasmid sequence in this assembly</p>
<ul>
<li>In the left hand panel: Blast: create/view BLAST search</li>
<li>Build blast database</li>
<li>Paste in the sequence of contig3.fasta</li>
<li>Run Blast search</li>
<li>There are two hits around a node (in this case, node 249).</li>
</ul>
</li>
<li>
<p>Go to the main Bandage window</p>
<ul>
<li>In the right hand panel, enter the node number.</li>
<li>Click &ldquo;Find nodes&rdquo;</li>
<li>This node is a circular contig in the graph, and is slightly longer (2329) than our contig3 (2229): this could be the plasmid.</li>
<li>Extract this node in fasta format: In the top panel, go to Output: Save selected node sequences; save as <fn>contig3b.fasta</fn></li>
</ul>
</li>
<li>
<p>Move this file back to the analysis folder.</p>
</li>
<li>Open this file in nano and change the header to &ldquo;&gt;contig3b&rdquo;, save.</li>
<li>Take the start of the sequence and see if it matches the end:</li>
</ul>
<div class="codehilite"><pre><span></span>head -n 10 contig3b.fasta &gt; contig3b.fa.head
makeblastdb -in contig3b.fasta -dbtype nucl
blastn -query contig3b.fa.head -db contig3b.fasta -evalue 1e-3 -dust no -out contig3b.bls
less contig3b.bls
</pre></div>


<ul>
<li>The first hit is against the start of the chromosome, as expected.</li>
<li>The last hit starts at position 2253; we will trim the plasmid to position 2252</li>
<li>Index and trim the contig3b.fa file:</li>
</ul>
<div class="codehilite"><pre><span></span>samtools faidx contig3b.fasta
samtools faidx contig3b.fasta contig3b:1-2252 &gt; contig3b.fa.trimmed
</pre></div>


<ul>
<li>Open this file in nano and change the header to &ldquo;&gt;contig3b&rdquo;, save.</li>
<li>We now have a trimmed contig3b.</li>
<li>Join all contigs:</li>
</ul>
<div class="codehilite"><pre><span></span>cat contigs_1_2.fasta contig3b.fa.trimmed &gt; genome.fasta
</pre></div>


<h3 id="correct_1">Correct</h3>
<div class="codehilite"><pre><span></span>bwa index genome.fasta
bwa mem -t 32 genome.fasta illumina_R1.fastq.gz illumina_R2.fastq.gz | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx genome.fasta
</pre></div>


<div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --verbose --threads 32
</pre></div>


<ul>
<li>Look at the <fn>pilon1.changes</fn> file.</li>
<li>Change the file name.</li>
</ul>
<div class="codehilite"><pre><span></span>cp pilon1.fasta assembly.fasta
</pre></div>


<ul>
<li>Look at the final assembly:</li>
</ul>
<div class="codehilite"><pre><span></span>infoseq assembly.fasta
</pre></div>


<ul>
<li>
<p>Assembly details:</p>
<ul>
<li>Chromosome: 2725222</li>
<li>Large plasmid: 25012</li>
<li>Small plasmid: 2252</li>
</ul>
</li>
</ul>
<h2 id="next">Next</h2>
<p><strong>Further analyses:</strong></p>
<ul>
<li>Annotate with Prokka.</li>
<li>Comparative genomics, e.g. with Roary.</li>
</ul>
<p><strong>Links:</strong></p>
<ul>
<li><a href="https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf">Details of bas.h5 files</a></li>
<li>Canu <a href="http://canu.readthedocs.io/en/stable/quick-start.html">manual</a> and <a href="https://github.com/marbl/canu">gitub repository</a></li>
<li>Circlator <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0849-0">article</a> and <a href="http://sanger-pathogens.github.io/circlator/">github repository</a></li>
<li>Pilon <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0112963">article</a> and <a href="https://github.com/broadinstitute/pilon/wiki">github repository</a></li>
<li>Notes on <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Finishing-Bacterial-Genomes">finishing</a> and <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Evaluating-Assemblies">evaluating</a> assemblies.</li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.5e60981f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>