



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/roary-galaxy/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.2.0">
    
    
      
        <title>Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.572ca0f0.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.8c900955.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#find-pan-genomes-using-roary-in-galaxy" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Find pan-genomes using Roary (in Galaxy)
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy-data/" title="Download and Delete Data in Galaxy" class="md-nav__link">
      Download and Delete Data in Galaxy
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant calling - bacteria" class="md-nav__link">
      Variant calling - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - eukaryotes" class="md-nav__link">
      RNA-seq - eukaryotes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - Bacterial RNA-seq" class="md-nav__link">
      Workflow - Bacterial RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-metagenomics/" title="Workflow - Metagenomics" class="md-nav__link">
      Workflow - Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../help/" title="Help!" class="md-nav__link">
      Help!
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run" title="Run" class="md-nav__link">
    Run
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-roary" title="Run Roary" class="md-nav__link">
    Run Roary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#what-does-roary-do" title="What does Roary do" class="md-nav__link">
    What does Roary do
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output" title="Output" class="md-nav__link">
    Output
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#summary-statistics" title="Summary statistics" class="md-nav__link">
    Summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#core-gene-alignment" title="Core gene alignment" class="md-nav__link">
    Core gene alignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-presenceabsence" title="Gene presence/absence" class="md-nav__link">
    Gene presence/absence
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#infer-phylogeny-using-core-gene-snps" title="Infer phylogeny using core gene snps" class="md-nav__link">
    Infer phylogeny using core gene snps
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vizualize-with-phandango" title="Vizualize with Phandango" class="md-nav__link">
    Vizualize with Phandango
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/roary-galaxy/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="find-pan-genomes-using-roary-in-galaxy">Find pan-genomes using Roary (in Galaxy)</h1>
<p>A concept in comparative microbial genomics is core and pan genomes. If we analyse DNA from several bacterial strains, we may want to know which genes they have in common and which are unique to some strains.</p>
<ul>
<li>
<p>The <em>core genome</em> is the group of genes shared by all strains in the clade of interest. Gene sequences are similar but not necessarily identical. &ldquo;Core genome SNPs&rdquo; are those SNPs found in the genes in the core genome; i.e. at a particular site, the nucleotide varies. We can use these SNPs to infer relationships between the strains.</p>
</li>
<li>
<p>The <em>accessory genome</em> is the group of genes that are not in all the strains. These genes may be in one or more strain.</p>
</li>
<li>
<p>The <em>pan genome</em> is the sum of the core and accessory genomes. That is, a combination of <em>all</em> the genes that are found in the clade of interest.</p>
</li>
</ul>
<p>Bacteria can horizontally-transfer genes to other bacteria via plasmids, and so their accessory genome can be large relative to those of eurkaryotes. Bacterial accessory genomes often house genes for drug resistance.</p>
<p>This tutorial demonstrates how to calculate the pan and core genomes of a set of input bacterial samples, using the tool <a href="https://sanger-pathogens.github.io/Roary/">Roary</a>.</p>
<h2 id="get-data">Get data</h2>
<p>Input:</p>
<ul>
<li>annotated genome in GFF3 format, per sample (<em>e.g.</em> output from Prokka)</li>
<li>samples must be from same species</li>
</ul>
<p>Our data:</p>
<ul>
<li>Five strains of <em>Staphylococcus aureus</em>, in .gff format.</li>
<li>Galaxy or swift location: ask your demonstrator or load your own data. </li>
</ul>
<!-- depends on data release policy-->

<h2 id="run">Run</h2>
<h3 id="run-roary">Run Roary</h3>
<ul>
<li>In Galaxy, go to <ss>Tools &rarr; NGS Analysis &rarr; Pan Genomes &rarr; Roary</ss>  </li>
<li>
<p>Set the following parameters (leave everything else unchanged):</p>
<ul>
<li><ss>Individual gff files or a dataset collection</ss>: <em>Individual</em></li>
<li><ss>select gff inputs to Roary</ss>: <em>Select all the .gff input files</em></li>
</ul>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
</ul>
<h3 id="what-does-roary-do">What does Roary do</h3>
<ul>
<li>converts coding sequences into protein sequences</li>
<li>clustered these protein sequences by several methods</li>
<li>further refines clusters into orthologous genes</li>
<li>for each sample, determines if gene is present/absent: produces <fn>gene_presence_absence.csv</fn></li>
<li>uses this gene p/a information to build a tree, using FastTree: produces <fn>accessory_binary_genes.fa.newick</fn></li>
<li>overall, calculates number of genes that are shared, and unique: produces <fn>summary_statistics.txt</fn></li>
<li>aligns the core genes (if option used, as above) for downstream analyses</li>
</ul>
<h2 id="output">Output</h2>
<p>There are three output files. </p>
<!-- (TBA: should be four, if newick tree available) -->

<h3 id="summary-statistics">Summary statistics</h3>
<p>Click on the eye icon. This shows a table of counts of shared genes (core genome) and total genes (pan genome).</p>
<p><img alt="summary stats" src="images/summary_stats.png" /></p>
<h3 id="core-gene-alignment">Core gene alignment</h3>
<p>Click on the disk icon under this file to download it.</p>
<h3 id="gene-presenceabsence">Gene presence/absence</h3>
<p>Click on the disk icon under this file to download it. Open and view in spreadsheet software.</p>
<ul>
<li>Column 3 shows the annotated gene name.</li>
<li>Column 4 shows the number of isolates that the gene was found it (in this case, ordered from 5 (all) to 1).</li>
</ul>
<!-- ### Newick TBA- need the accessory_binary_genes.fa.newick (not just .fa) output file -->

<h2 id="infer-phylogeny-using-core-gene-snps">Infer phylogeny using core gene snps</h2>
<!-- [Optional step. Or use .nwk tree from gene p/a] -->

<p>Roary has produced an alignment of the core genes. We can use this alignment to infer a phylogenetic tree of the isolates.</p>
<ul>
<li>In Galaxy, go to <ss>Tools &rarr; NGS Analysis &rarr; NGS:Phylogenetics &rarr; Phylogeneitc reconstruction with RaXML</ss>  </li>
<li>
<p>Set the following parameters (leave everything else unchanged):</p>
<ul>
<li><ss>Source file</ss>: <fn>Roary on data x, data x, and others Core Gene Alignment</fn></li>
<li><ss>Model Type</ss>: <em>Nucleotide</em></li>
<li><ss>Substitution Model</ss>: <em>GTRGAMMA</em></li>
</ul>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
</ul>
<p>There are six output files.</p>
<p>Click on <ss>Result</ss>. Under the file, click on the &ldquo;Visualize&rdquo; icon (a graph), then choose PhyloViz.</p>
<p><img alt="viz button" src="images/viz_button.png" /></p>
<!--[Need to rename these isolates instead of temp0 etc]-->

<ul>
<li>These isolates are all very closely related and so the structure of the tree is narrow.</li>
<li>To expand, go to the right hand box for &ldquo;Phyloviz Settings&rdquo;. Change the Phylogenetic Spacing to 2500 and the Vertical Spacing to 30.</li>
</ul>
<p>To return to the main Galaxy screen, go to the top panel and click on &ldquo;Analyze Data&rdquo;.</p>
<p>Click on the disk icon under the <fn>Results</fn> file to download.</p>
<p>Re-name with the file extension <fn>.tree</fn></p>
<h2 id="vizualize-with-phandango">Vizualize with Phandango</h2>
<p>If not done already, copy these files to your local computer:</p>
<ul>
<li>
<p>The <fn>raxml.tree</fn> (or the <fn>accessory_binary_genes.fa.newick</fn>).</p>
</li>
<li>
<p><fn>gene_presence_absence.csv</fn></p>
</li>
</ul>
<p>Go to <a href="http://phandango.net">http://phandango.net</a></p>
<ul>
<li>drag and drop the two files onto the landing page.</li>
<li>view the tree of samples and their core and pan genomes</li>
<li>each blue coloured column is a gene: genes are present or absent in each isolate</li>
<li>the core genes are shared by all isolates</li>
</ul>
<p><img alt="phandango" src="images/phandango.png" /></p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b41f3d20.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>