



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for using Galaxy Australia">
      
      
        <link rel="canonical" href="https://galaxy-au-training.github.io/tutorials/modules/snippy/">
      
      
        <meta name="author" content="Anna Syme, Gareth Price, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.4, mkdocs-material-2.9.0">
    
    
      
        <title>Variant calling - bacteria - Galaxy Australia Training</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.4f4f49e8.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#variant-calling-with-snippy" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://galaxy-au-training.github.io/tutorials/" title="Galaxy Australia Training" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Galaxy Australia Training
              </span>
              <span class="md-header-nav__topic">
                Variant calling - bacteria
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Galaxy Australia Training
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/Galaxy-Au-training/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Galaxy-Australia-training/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Welcome" class="md-nav__link">
      Welcome
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../galaxy/" title="Get started" class="md-nav__link">
      Get started
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../intro/" title="Learn key tasks" class="md-nav__link">
      Learn key tasks
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly" class="md-nav__link">
      Genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Variant calling - bacteria
      </label>
    
    <a href="./" title="Variant calling - bacteria" class="md-nav__link md-nav__link--active">
      Variant calling - bacteria
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-shared-data" title="Use shared data" class="md-nav__link">
    Use shared data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#or-import-from-the-web" title="Or, import from the web" class="md-nav__link">
    Or, import from the web
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shorten-file-names" title="Shorten file names" class="md-nav__link">
    Shorten file names
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#call-variants-with-snippy" title="Call variants with Snippy" class="md-nav__link">
    Call variants with Snippy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-snippy-output" title="Examine Snippy output" class="md-nav__link">
    Examine Snippy output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#view-snippy-output-in-jbrowse" title="View Snippy output in JBrowse" class="md-nav__link">
    View Snippy output in JBrowse
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extension-exercise" title="Extension exercise" class="md-nav__link">
    Extension exercise
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-this-history-in-galaxy" title="See this history in Galaxy" class="md-nav__link">
    See this history in Galaxy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#whats-next" title="What's next?" class="md-nav__link">
    What's next?
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variants/" title="Variant calling - human" class="md-nav__link">
      Variant calling - human
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../abricate/" title="Antibiotic-resistant genes" class="md-nav__link">
      Antibiotic-resistant genes
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mlst/" title="Using MLST" class="md-nav__link">
      Using MLST
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification" class="md-nav__link">
      Sample identification
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../dge/" title="RNA-seq - bacteria" class="md-nav__link">
      RNA-seq - bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna-humans/" title="RNA-seq - humans" class="md-nav__link">
      RNA-seq - humans
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metagenomics/metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows - introduction" class="md-nav__link">
      Galaxy workflows - introduction
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-assembly/" title="Workflow - genome assembly" class="md-nav__link">
      Workflow - genome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-annotation/" title="Workflow - genome annotation" class="md-nav__link">
      Workflow - genome annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-variant-calling/" title="Workflow - variant calling, bacteria" class="md-nav__link">
      Workflow - variant calling, bacteria
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../workflow-rna-seq/" title="Workflow - RNA-seq" class="md-nav__link">
      Workflow - RNA-seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../shortreads/" title="Short read data" class="md-nav__link">
      Short read data
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../refseq/" title="Reference genomes" class="md-nav__link">
      Reference genomes
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#get-data" title="Get data" class="md-nav__link">
    Get data
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-shared-data" title="Use shared data" class="md-nav__link">
    Use shared data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#or-import-from-the-web" title="Or, import from the web" class="md-nav__link">
    Or, import from the web
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shorten-file-names" title="Shorten file names" class="md-nav__link">
    Shorten file names
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#call-variants-with-snippy" title="Call variants with Snippy" class="md-nav__link">
    Call variants with Snippy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-snippy-output" title="Examine Snippy output" class="md-nav__link">
    Examine Snippy output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#view-snippy-output-in-jbrowse" title="View Snippy output in JBrowse" class="md-nav__link">
    View Snippy output in JBrowse
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extension-exercise" title="Extension exercise" class="md-nav__link">
    Extension exercise
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-this-history-in-galaxy" title="See this history in Galaxy" class="md-nav__link">
    See this history in Galaxy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#whats-next" title="What's next?" class="md-nav__link">
    What's next?
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/Galaxy-Au-training/tutorials/edit/master/docs/modules/snippy/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="variant-calling-with-snippy">Variant calling with Snippy</h1>
<p>Variant calling is the process of identifying differences between two genome samples.</p>
<ul>
<li>Differences are usually limited to single nucleotide polymorphisms (SNPs) and small insertions and deletions (indels).</li>
<li>Larger structural variation such as inversions, duplications and large deletions are not typically covered by &ldquo;variant calling&rdquo;.</li>
<li>In this tutorial, we will use the tool &ldquo;Snippy&rdquo; (link to Snippy is <a href="https://github.com/tseemann/snippy">here</a>).</li>
<li>Snippy uses a tool to align the reads to a reference genome, and another tool to decide (&ldquo;call&rdquo;) if the discrepancies are real variants.</li>
</ul>
<p><fn><strong>New to Galaxy?</strong> First try the <a href="../galaxy/">introduction</a> and then learn some <a href="../intro/">key tasks</a></fn></p>
<!--
## Learning Objectives

1. Find variants between a reference genome and a set of reads
2. Visualise the SNP in context of the reads aligned to the genome
3. Determine the effect of those variants on genomic features
4. Understand if the SNP is potentially affecting the phenotype
-->

<h2 id="get-data">Get data</h2>
<ul>
<li>Log in to your Galaxy instance (for example, Galaxy Australia, <a href="https://usegalaxy.org.au/">usegalaxy.org.au</a>).</li>
</ul>
<h3 id="use-shared-data">Use shared data</h3>
<p>If you are using Galaxy Australia, you can import the data from a shared data library.</p>
<p>In the top menu bar, go to <ss>Shared Data</ss>.</p>
<ul>
<li>Click on <ss>Data Libraries</ss>.</li>
<li>Click on <fn>Galaxy Australia Training Material: Variant Calling: Microbial Variant Calling</fn>.</li>
<li>Tick the boxes next to the five files.</li>
<li>Click the <ss>To History</ss> button, select <em>As Datasets</em>.</li>
<li>Name a new history and click <ss>Import</ss>.</li>
<li>In the top menu bar, click <ss>Analyze Data</ss>.</li>
<li>You should now have five files in your current history.</li>
</ul>
<h3 id="or-import-from-the-web">Or, import from the web</h3>
<p><em>Only follow this step if unable to load the data files from shared data, as described above</em>.</p>
<ul>
<li>In a new browser tab, go to this webpage:</li>
</ul>
<p><a href="https://doi.org/10.5281/zenodo.582600"><img alt="DOI" src="https://zenodo.org/badge/DOI/10.5281/zenodo.582600.svg" /></a></p>
<ul>
<li>Find the file called <fn>mutant_R1.fastq</fn></li>
<li>Right click on file name: select &ldquo;copy link address&rdquo;</li>
<li>In Galaxy, go to <ss>Get Data</ss> and then Upload File</li>
<li>Click <ss>Paste/Fetch data</ss></li>
<li>A box will appear: paste in link address</li>
<li>Click <ss>Start</ss></li>
<li>Click <ss>Close</ss></li>
<li>The file will now appear in the top of your history panel.</li>
<li>Shorten the file name if you wish. (Click the pencil icon).</li>
</ul>
<p>Repeat these steps for the other set of reads (reverse reads) called <fn>mutant_R2.fq</fn>:</p>
<ul>
<li>In a new browser tab, go to this webpage:</li>
</ul>
<p><a href="https://doi.org/10.5281/zenodo.582600"><img alt="DOI" src="https://zenodo.org/badge/DOI/10.5281/zenodo.582600.svg" /></a></p>
<ul>
<li>Find the file called <fn>mutant_R2.fastq</fn></li>
<li>Right click on file name: select &ldquo;copy link address&rdquo;</li>
<li>In Galaxy, go to <ss>Get Data</ss> and then Upload File</li>
<li>Click <ss>Paste/Fetch data</ss></li>
<li>A box will appear: paste in link address</li>
<li>Click <ss>Start</ss></li>
<li>Click <ss>Close</ss></li>
<li>The file will now appear in the top of your history panel.</li>
<li>Shorten the file name if you wish.</li>
</ul>
<p>Repeat these steps for three other files: <fn>wildtype.gbk</fn>, <fn>wildtype.gff</fn> and <fn>wildtype.fna</fn>.</p>
<h3 id="shorten-file-names">Shorten file names</h3>
<ul>
<li>Click on the pencil icon next to the file name.</li>
<li>In the centre Galaxy panel, click in the box under <ss>Name</ss></li>
<li>Shorten the file name.</li>
<li>Then click <ss>Save</ss></li>
</ul>
<h2 id="call-variants-with-snippy">Call variants with Snippy</h2>
<p>Go to the Tool panel and search for &ldquo;snippy&rdquo; in the search box.</p>
<ul>
<li>Click on <ss>snippy</ss></li>
</ul>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li>For <ss>Reference type</ss> select <em>Genbank</em>.</li>
<li>Then for <ss>Reference Genbank</ss> choose the <fn>wildtype.gbk</fn> file.</li>
<li>For <ss>Single or Paired-end reads</ss> choose <em>Paired</em>.</li>
<li>Then choose the first set of reads, <fn>mutant_R1.fastq</fn> and second set of reads, <fn>mutant_R2.fastq</fn>.</li>
<li>For <ss>Cleanup the non-snp output files</ss> select <em>No</em>. The box will be a darker grey.</li>
</ul>
<p>Your tool interface should look like this:</p>
<p><img alt="Snippy interface" src="./images/interface.png" /></p>
<ul>
<li>Click <ss>Execute</ss>.</li>
</ul>
<h2 id="examine-snippy-output">Examine Snippy output</h2>
<!-- First, enable "Scratchbook" in Galaxy - this allows you to view several windows simultaneously. Click on the squares:

![scratchbook icon](images/scratchbook.png)

-->

<p>From Snippy, there are 10 output files in various formats.</p>
<ul>
<li>Go to the <strong>table</strong> file called <fn>snippy on data XX, data XX and data XX table</fn> and click on the eye icon.</li>
<li>(The XX refers to the number that Galaxy has given your files.)</li>
<li>We can see a list of variants. Look in column 3 to see which types the variants are, such as a SNP or a deletion.</li>
<li>Look at the third variant called. This is a T&rarr;A mutation, causing a stop codon. Look at column 14: the product of this gene is a methicillin resistance protein. Methicillin is an antibiotic. What might be the result of such a mutation? <!--[add a hint/info box]--></li>
</ul>
<h2 id="view-snippy-output-in-jbrowse">View Snippy output in JBrowse</h2>
<ul>
<li>
<p>Go to the Galaxy tools panel, and use the search box at the top to search for &ldquo;JBrowse&rdquo;.</p>
</li>
<li>
<p>Under <ss>Reference genome to display</ss> choose <em>Use a genome from history</em>.</p>
</li>
<li>
<p>Under <ss>Select the reference genome</ss> choose <fn>wildtype.fna</fn>. This sequence will be the reference against which annotations are displayed.</p>
</li>
<li>
<p>For <ss>Produce a Standalone Instance</ss> select <em>Yes</em>.</p>
</li>
<li>
<p>For <ss>Genetic Code</ss> choose <em>11: The Bacterial, Archaeal and Plant Plastid Code</em>.</p>
</li>
<li>
<p>Under <ss>JBrowse-in-Galaxy Action</ss> choose <em>New JBrowse Instance</em>.</p>
</li>
<li>
<p>We will now set up three different tracks - these are datasets displayed underneath the reference sequence (which is displayed as nucleotides in FASTA format). We will choose to display the sequence reads (the .bam file), the variants found by snippy (the .gff file) and the annotated reference genome (the wildtype.gff)</p>
</li>
</ul>
<p><em>Track 1 - sequence reads</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss></li>
<li>For <ss>Track Cateogry</ss> name it &ldquo;sequence reads&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>BAM Pileups</em></li>
<li>For <ss>BAM Track Data</ss> select <fn>the snippy bam file</fn></li>
<li>For <ss>Autogenerate SNP Track</ss> select <em>Yes</em></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 2 - variants</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss>Track Category</ss> name it &ldquo;variants&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>the snippy snps gff file</fn></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 3 - annotated reference</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss> Track Category</ss> name it &ldquo;annotated reference&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>wildtype.gff</fn></li>
<li>Under <ss>JBrowse Track Type[Advanced]</ss> select <em>Canvas Features</em>.</li>
<li>Click on <ss>JBrowse Styling Options <Advanced]</ss></li>
<li>Under <ss>JBrowse style.label</ss> add in the word <em>product</em>.</li>
<li>Under <ss>JBrowse style.description</ss> check that it says <em>product,name,id</em>.</li>
<li>
<p>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</p>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
</ul>
<p>A new file will be created, called <fn>JBrowse on data XX and data XX - Complete</fn>. (This may take some time. If you would like to see a completed history for this tutorial, instructions are in the next section.)</p>
<ul>
<li>
<p>Click on the eye icon next to the file name. The JBrowse window will appear in the centre Galaxy panel.</p>
</li>
<li>
<p>On the left, tick boxes to display the tracks</p>
</li>
<li>
<p>Use the minus button to zoom out to see:</p>
<ul>
<li>sequence reads and their coverage (the grey graph)</li>
</ul>
</li>
<li>
<p>Use the plus button to zoom in to see:</p>
<ul>
<li>probable real variants (a whole column of snps)</li>
<li>probable errors (single one here and there)</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse1.png" /></p>
<ul>
<li>In the coordinates box, type in <em>47299</em> and then <ss>Go</ss> to see the position of the SNP discussed above.<ul>
<li>the correct codon at this position is TGT, coding for the amino acid Cysteine, in the middle row of the amino acid translations.</li>
<li>the mutation of T &rarr; A turns this triplet into TGA, a stop codon.</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse2.png" />    </p>
<h2 id="extension-exercise">Extension exercise</h2>
<p>Let&rsquo;s repeat this tutorial using other data.</p>
<ul>
<li>We will investigate the bacteria <em>Mycoplasma synoviae</em> as it has a small genome and will not take too long to analyse.</li>
<li>We want to find out if our sequencing reads vary from a reference genome.</li>
<li>
<p>To get some sequencing reads, follow the instructions <a href="../shortreads/">here</a>.</p>
<ul>
<li>This will give us paired-end sequencing reads (two files).</li>
</ul>
</li>
<li>
<p>To obtain a reference genome, follow the instructions <a href="../refseq/">here</a> including the Prokka step.</p>
<ul>
<li>This will give us 12 output files from Prokka.</li>
<li>We need three of these files: the genbank file  <fn>Prokka.gbk</fn>, the fasta file <fn>Prokka.fna</fn>, and the gff file <fn>Prokka.gff</fn>.</li>
<li>Galaxy has interpreted the genbank file as a text file. Click on the pencil icon next to the genbank file; click Datatypes, and change the datatype to &ldquo;genbank&rdquo;.</li>
</ul>
</li>
<li>
<p>Run snippy on the sequencing reads.</p>
<ul>
<li>For the reference genome, use the <fn>Prokka:gbk</fn> file.</li>
</ul>
</li>
<li>
<p>Snippy has mapped the sequencing reads to the reference genome and produced a BAM file.</p>
<ul>
<li>This is a very large file so we will sub-sample it.</li>
<li>In the tool panel, search for &ldquo;downsample&rdquo;. Click on the &ldquo;Downsample SAM/BAM&rdquo; tool.</li>
<li>The input file is the snippy BAM file.</li>
<li>Set the probability to 0.01</li>
</ul>
</li>
<li>
<p>Run JBrowse to produce a visualization of the called variants.</p>
<ul>
<li>For the reference genome, use the <fn>Prokka.fna</fn> file.</li>
<li>As we did above, set up three &ldquo;Tracks&rdquo; to view underneath this reference genome nucleotide sequence: The <fn>downsampled.bam</fn> file showing the reads mapped to the reference, the <fn>snippy.gff</fn> file showing the snps (the variants), and the annotated reference genome <fn>Prokka.gff</fn>.</li>
</ul>
</li>
</ul>
<p>You should now have a JBrowse file showing the results of variant calling on these reads of the bacteria <em>Mycoplasma synoviae</em>. Can you find any variants?</p>
<h2 id="see-this-history-in-galaxy">See this history in Galaxy</h2>
<p>If you want to see this Galaxy history without performing the steps above:</p>
<ul>
<li>Log in to Galaxy Australia: <a href="https://usegalaxy.org.au/">https://usegalaxy.org.au/</a></li>
<li>Go to <ss>Shared Data</ss></li>
<li>Click <ss>Histories</ss></li>
<li>Click <fn>Completed-variant-calling-analysis</fn></li>
<li>Click <ss>Import</ss> (at the top right corner)</li>
<li>The analysis should now be showing as your current history.</li>
</ul>
<h2 id="whats-next">What&rsquo;s next?</h2>
<p>To use the tutorials on this website:</p>
<ul>
<li>&#8592; see the list in the left hand panel</li>
<li>&#8598; or, click the <strong>menu button</strong> (three horizontal bars) in the top left of the page</li>
</ul>
<p>You can find more tutorials at the Galaxy Training Network:</p>
<ul>
<li><a href="http://galaxyproject.github.io/training-material/">http://galaxyproject.github.io/training-material/</a></li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../prokka/" title="Genome annotation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Genome annotation
              </span>
            </div>
          </a>
        
        
          <a href="../variants/" title="Variant calling - human" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Variant calling - human
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2018-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../assets/fonts/font-awesome.css">
    
      <a href="" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/Galaxy-Au-training/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.a59e2a89.js"></script>
      
      <script>app.initialize({version:"0.17.4",url:{base:"../.."}})</script>
      
    
    
      
    
  </body>
</html>